// vue-tailwind-datepicker/dist/entry.58a71fc4.mjs
import { openBlock as K, createElementBlock as J, createElementVNode as L, withDirectives as ye, vShow as xe, toDisplayString as ne, Fragment as ke, renderList as Ee, inject as ue, createVNode as se, TransitionGroup as kt, withCtx as De, normalizeClass as Pe, Transition as tt, unref as G, createCommentVNode as Oe, withModifiers as $e, cloneVNode as $t, h as _e, provide as ie, ref as ee, onMounted as jt, watchEffect as be, computed as de, defineComponent as Ye, reactive as Mt, nextTick as Ae, watch as St, isProxy as Dt, createBlock as Ie, renderSlot as Ot, mergeProps as Vt, vModelText as Pt } from "vue";
const _t = { class: "flex justify-between items-center px-2 py-1.5" }, Ct = { class: "flex-shrink-0" }, Yt = { class: "inline-flex rounded-full" }, Tt = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, At = ["d"], Lt = { class: "px-1.5 space-x-1.5 flex flex-1" }, Bt = { class: "flex-1 flex rounded-md" }, Et = ["textContent"], Ft = { class: "flex-1 flex rounded-md" }, Nt = ["textContent"], Ht = { class: "flex-shrink-0" }, Rt = { class: "inline-flex rounded-full" }, It = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Ut = ["d"], Ke = {
  __name: "Header",
  props: {
    asPrevOrNext: Boolean,
    panel: Object,
    calendar: Object
  },
  setup(t) {
    return (a, e) => (K(), J("div", _t, [
      L("div", Ct, [
        ye(L("span", Yt, [
          L("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: e[0] || (e[0] = (l) => t.panel.calendar ? t.calendar.onPrevious() : t.calendar.onPreviousYear())
          }, [
            (K(), J("svg", Tt, [
              L("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: t.panel.calendar ? "M15 19l-7-7 7-7" : "M11 19l-7-7 7-7m8 14l-7-7 7-7"
              }, null, 8, At)
            ]))
          ])
        ], 512), [
          [xe, t.panel.calendar || t.panel.year]
        ])
      ]),
      L("div", Lt, [
        L("span", Bt, [
          L("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            textContent: ne(t.calendar.month),
            onClick: e[1] || (e[1] = (l) => t.calendar.openMonth())
          }, null, 8, Et)
        ]),
        L("span", Ft, [
          L("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            textContent: ne(t.calendar.year),
            onClick: e[2] || (e[2] = (l) => t.calendar.openYear())
          }, null, 8, Nt)
        ])
      ]),
      L("div", Ht, [
        ye(L("span", Rt, [
          L("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: e[3] || (e[3] = (l) => t.panel.calendar ? t.calendar.onNext() : t.calendar.onNextYear())
          }, [
            (K(), J("svg", It, [
              L("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: t.panel.calendar ? "M9 5l7 7-7 7" : "M13 5l7 7-7 7M5 5l7 7-7 7"
              }, null, 8, Ut)
            ]))
          ])
        ], 512), [
          [xe, t.panel.calendar || t.panel.year]
        ])
      ])
    ]));
  }
}, zt = { class: "flex flex-wrap mt-1.5" }, Wt = { class: "flex rounded-md mt-1.5" }, Zt = ["textContent", "onClick"], Je = {
  __name: "Month",
  props: {
    months: Array
  },
  emits: ["update:month"],
  setup(t, { emit: a }) {
    return (e, l) => (K(), J("div", zt, [
      (K(!0), J(ke, null, Ee(t.months, (n, i) => (K(), J("div", {
        key: i,
        class: "w-1/2 px-0.5"
      }, [
        L("span", Wt, [
          L("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            textContent: ne(n),
            onClick: (V) => a("update:month", i)
          }, null, 8, Zt)
        ])
      ]))), 128))
    ]));
  }
}, Gt = { class: "grid grid-cols-7 py-2 mt-0.5" }, Kt = ["textContent"], qe = {
  __name: "Week",
  props: {
    weeks: Array
  },
  setup(t) {
    return (a, e) => (K(), J("div", Gt, [
      (K(!0), J(ke, null, Ee(t.weeks, (l, n) => (K(), J("div", {
        key: n,
        class: "text-vtd-secondary-500 text-xs 2xl:text-sm tracking-wide font-medium text-center cursor-default dark:text-vtd-secondary-400"
      }, [
        L("span", {
          textContent: ne(l)
        }, null, 8, Kt)
      ]))), 128))
    ]));
  }
}, Jt = { class: "flex flex-wrap" }, qt = { class: "flex rounded-md mt-1.5" }, Xt = ["textContent", "onClick"], Xe = {
  __name: "Year",
  props: {
    asPrevOrNext: Boolean,
    years: Array
  },
  emits: ["update:year"],
  setup(t, { emit: a }) {
    return (e, l) => (K(), J("div", Jt, [
      (K(!0), J(ke, null, Ee(t.years, (n, i) => (K(), J("div", {
        key: i,
        class: "w-1/2 px-0.5"
      }, [
        L("span", qt, [
          L("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            textContent: ne(n),
            onClick: (V) => a("update:year", n, t.asPrevOrNext)
          }, null, 8, Xt)
        ])
      ]))), 128))
    ]));
  }
}, Qt = { class: "grid grid-cols-7 gap-y-0.5 my-1" }, er = ["data-tooltip"], tr = ["disabled", "onClick", "onMouseenter", "onFocusin", "textContent", "data-date"], Qe = {
  __name: "Calendar",
  props: {
    asPrevOrNext: Boolean,
    calendar: Object,
    weeks: Array,
    asRange: Boolean
  },
  emits: ["update:date"],
  setup(t, { emit: a }) {
    const e = ue("isBetweenRange"), l = ue("betweenRangeClasses"), n = ue("datepickerClasses"), i = ue("atMouseOver");
    return (V, c) => (K(), J("div", Qt, [
      se(kt, {
        "enter-from-class": "opacity-0",
        "enter-to-class": "opacity-100",
        "enter-active-class": "transition-opacity ease-out duration-300",
        "leave-active-class": "transition-opacity ease-in duration-200",
        "leave-from-class": "opacity-100",
        "leave-to-class": "opacity-0"
      }, {
        default: De(() => [
          (K(!0), J(ke, null, Ee(t.calendar.date(), (f, h) => (K(), J("div", {
            key: h,
            class: Pe(["relative", { "vtd-tooltip": t.asRange && f.duration() }]),
            "data-tooltip": `${f.duration()}`
          }, [
            se(tt, {
              "enter-from-class": "opacity-0",
              "enter-to-class": "opacity-100",
              "enter-active-class": "transition-opacity ease-out duration-200",
              "leave-active-class": "transition-opacity ease-in duration-150",
              "leave-from-class": "opacity-100",
              "leave-to-class": "opacity-0"
            }, {
              default: De(() => [
                G(e)(f) || f.hovered() ? (K(), J("span", {
                  key: 0,
                  class: Pe(["absolute bg-vtd-primary-100 bg-opacity-60 dark:bg-vtd-secondary-700 dark:bg-opacity-50", G(l)(f)])
                }, null, 2)) : Oe("", !0)
              ]),
              _: 2
            }, 1024),
            L("button", {
              type: "button",
              class: Pe(["vtd-datepicker-date relative w-12 h-12 lg:w-10 lg:h-10 flex justify-center items-center text-xs 2xl:text-sm focus:outline-none", [G(n)(f), t.asRange ? "transition-all" : "transition-colors"]]),
              disabled: f.disabled || f.inRange(),
              onClick: ($) => a("update:date", f, t.asPrevOrNext),
              onMouseenter: ($) => G(i)(f),
              onFocusin: ($) => G(i)(f),
              textContent: ne(f.date()),
              "data-date": f.toDate()
            }, null, 42, tr)
          ], 10, er))), 128))
        ]),
        _: 1
      })
    ]));
  }
}, rr = {
  key: 0,
  class: "relative w-full border-t border-b-0 sm:border-t-0 sm:border-b lg:border-b-0 lg:border-r border-black/[.1] order-last sm:order-none dark:border-vtd-secondary-700/[1] sm:mt-1 lg:mr-1 sm:mb-1 lg:mb-0 sm:mx-1 lg:mx-0"
}, or = {
  key: 0,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, ar = ["onClick", "textContent"], nr = {
  key: 1,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, sr = {
  __name: "Shortcut",
  props: {
    shortcuts: [Boolean, Function],
    close: Function,
    asRange: Boolean,
    asSingle: Boolean,
    i18n: Object
  },
  setup(t) {
    const a = t, e = ue("setToToday"), l = ue("setToYesterday"), n = ue("setToLastDay"), i = ue("setToThisMonth"), V = ue("setToLastMonth"), c = ue("setToCustomShortcut"), f = () => typeof a.shortcuts == "function" ? a.shortcuts() : !1;
    return (h, $) => a.asRange && !a.asSingle ? (K(), J("div", rr, [
      f() ? (K(), J("ol", or, [
        (K(!0), J(ke, null, Ee(f(), (b, w) => (K(), J("li", { key: w }, [
          L("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $e((A) => G(c)(b, t.close), ["prevent"]),
            textContent: ne(b.label)
          }, null, 8, ar)
        ]))), 128))
      ])) : (K(), J("ol", nr, [
        L("li", null, [
          L("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $[0] || ($[0] = $e((b) => G(e)(t.close), ["prevent"]))
          }, ne(a.i18n.today), 1)
        ]),
        L("li", null, [
          L("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $[1] || ($[1] = $e((b) => G(l)(t.close), ["prevent"]))
          }, ne(a.i18n.yesterday), 1)
        ]),
        L("li", null, [
          L("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $[2] || ($[2] = $e((b) => G(n)(7, t.close), ["prevent"]))
          }, ne(a.i18n.past(7)), 1)
        ]),
        L("li", null, [
          L("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $[3] || ($[3] = $e((b) => G(n)(30, t.close), ["prevent"]))
          }, ne(a.i18n.past(30)), 1)
        ]),
        L("li", null, [
          L("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $[4] || ($[4] = $e((b) => G(i)(t.close), ["prevent"]))
          }, ne(a.i18n.currentMonth), 1)
        ]),
        L("li", null, [
          L("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: $[5] || ($[5] = $e((b) => G(V)(t.close), ["prevent"]))
          }, ne(a.i18n.pastMonth), 1)
        ])
      ]))
    ])) : Oe("", !0);
  }
};
function we(t, a, ...e) {
  if (t in a) {
    let n = a[t];
    return typeof n == "function" ? n(...e) : n;
  }
  let l = new Error(`Tried to handle "${t}" but there is no handler defined. Only defined handlers are: ${Object.keys(a).map((n) => `"${n}"`).join(", ")}.`);
  throw Error.captureStackTrace && Error.captureStackTrace(l, we), l;
}
var Le = ((t) => (t[t.None = 0] = "None", t[t.RenderStrategy = 1] = "RenderStrategy", t[t.Static = 2] = "Static", t))(Le || {}), lr = ((t) => (t[t.Unmount = 0] = "Unmount", t[t.Hidden = 1] = "Hidden", t))(lr || {});
function Te({ visible: t = !0, features: a = 0, ourProps: e, theirProps: l, ...n }) {
  var i;
  let V = ir(l, e), c = Object.assign(n, { props: V });
  if (t || a & 2 && V.static)
    return Ue(c);
  if (a & 1) {
    let f = (i = V.unmount) == null || i ? 0 : 1;
    return we(f, { [0]() {
      return null;
    }, [1]() {
      return Ue({ ...n, props: { ...V, hidden: !0, style: { display: "none" } } });
    } });
  }
  return Ue(c);
}
function Ue({ props: t, attrs: a, slots: e, slot: l, name: n }) {
  var i;
  let { as: V, ...c } = ot(t, ["unmount", "static"]), f = (i = e.default) == null ? void 0 : i.call(e, l), h = {};
  if (l) {
    let $ = !1, b = [];
    for (let [w, A] of Object.entries(l))
      typeof A == "boolean" && ($ = !0), A === !0 && b.push(w);
    $ && (h["data-headlessui-state"] = b.join(" "));
  }
  if (V === "template") {
    if (f = rt(f), Object.keys(c).length > 0 || Object.keys(a).length > 0) {
      let [$, ...b] = f != null ? f : [];
      if (!ur($) || b.length > 0)
        throw new Error(['Passing props on "template"!', "", `The current component <${n} /> is rendering a "template".`, "However we need to passthrough the following props:", Object.keys(c).concat(Object.keys(a)).sort((w, A) => w.localeCompare(A)).map((w) => `  - ${w}`).join(`
`), "", "You can apply a few solutions:", ['Add an `as="..."` prop, to ensure that we render an actual element instead of a "template".', "Render a single element as the child so that we can forward the props onto that element."].map((w) => `  - ${w}`).join(`
`)].join(`
`));
      return $t($, Object.assign({}, c, h));
    }
    return Array.isArray(f) && f.length === 1 ? f[0] : f;
  }
  return _e(V, Object.assign({}, c, h), f);
}
function rt(t) {
  return t.flatMap((a) => a.type === ke ? rt(a.children) : [a]);
}
function ir(...t) {
  if (t.length === 0)
    return {};
  if (t.length === 1)
    return t[0];
  let a = {}, e = {};
  for (let l of t)
    for (let n in l)
      n.startsWith("on") && typeof l[n] == "function" ? (e[n] != null || (e[n] = []), e[n].push(l[n])) : a[n] = l[n];
  if (a.disabled || a["aria-disabled"])
    return Object.assign(a, Object.fromEntries(Object.keys(e).map((l) => [l, void 0])));
  for (let l in e)
    Object.assign(a, { [l](n, ...i) {
      let V = e[l];
      for (let c of V) {
        if (n instanceof Event && n.defaultPrevented)
          return;
        c(n, ...i);
      }
    } });
  return a;
}
function ot(t, a = []) {
  let e = Object.assign({}, t);
  for (let l of a)
    l in e && delete e[l];
  return e;
}
function ur(t) {
  return t == null ? !1 : typeof t.type == "string" || typeof t.type == "object" || typeof t.type == "function";
}
let dr = 0;
function cr() {
  return ++dr;
}
function Ce() {
  return cr();
}
var ge = ((t) => (t.Space = " ", t.Enter = "Enter", t.Escape = "Escape", t.Backspace = "Backspace", t.Delete = "Delete", t.ArrowLeft = "ArrowLeft", t.ArrowUp = "ArrowUp", t.ArrowRight = "ArrowRight", t.ArrowDown = "ArrowDown", t.Home = "Home", t.End = "End", t.PageUp = "PageUp", t.PageDown = "PageDown", t.Tab = "Tab", t))(ge || {});
function U(t) {
  var a;
  return t == null || t.value == null ? null : (a = t.value.$el) != null ? a : t.value;
}
let at = Symbol("Context");
var Be = ((t) => (t[t.Open = 0] = "Open", t[t.Closed = 1] = "Closed", t))(Be || {});
function nt() {
  return ue(at, null);
}
function fr(t) {
  ie(at, t);
}
function et(t, a) {
  if (t)
    return t;
  let e = a != null ? a : "button";
  if (typeof e == "string" && e.toLowerCase() === "button")
    return "button";
}
function mr(t, a) {
  let e = ee(et(t.value.type, t.value.as));
  return jt(() => {
    e.value = et(t.value.type, t.value.as);
  }), be(() => {
    var l;
    e.value || !U(a) || U(a) instanceof HTMLButtonElement && !((l = U(a)) != null && l.hasAttribute("type")) && (e.value = "button");
  }), e;
}
const He = typeof window > "u" || typeof document > "u";
function Fe(t) {
  if (He)
    return null;
  if (t instanceof Node)
    return t.ownerDocument;
  if (t != null && t.hasOwnProperty("value")) {
    let a = U(t);
    if (a)
      return a.ownerDocument;
  }
  return document;
}
let We = ["[contentEditable=true]", "[tabindex]", "a[href]", "area[href]", "button:not([disabled])", "iframe", "input:not([disabled])", "select:not([disabled])", "textarea:not([disabled])"].map((t) => `${t}:not([tabindex='-1'])`).join(",");
var je = ((t) => (t[t.First = 1] = "First", t[t.Previous = 2] = "Previous", t[t.Next = 4] = "Next", t[t.Last = 8] = "Last", t[t.WrapAround = 16] = "WrapAround", t[t.NoScroll = 32] = "NoScroll", t))(je || {}), pr = ((t) => (t[t.Error = 0] = "Error", t[t.Overflow = 1] = "Overflow", t[t.Success = 2] = "Success", t[t.Underflow = 3] = "Underflow", t))(pr || {}), vr = ((t) => (t[t.Previous = -1] = "Previous", t[t.Next = 1] = "Next", t))(vr || {});
function st(t = document.body) {
  return t == null ? [] : Array.from(t.querySelectorAll(We));
}
var Ge = ((t) => (t[t.Strict = 0] = "Strict", t[t.Loose = 1] = "Loose", t))(Ge || {});
function lt(t, a = 0) {
  var e;
  return t === ((e = Fe(t)) == null ? void 0 : e.body) ? !1 : we(a, { [0]() {
    return t.matches(We);
  }, [1]() {
    let l = t;
    for (; l !== null; ) {
      if (l.matches(We))
        return !0;
      l = l.parentElement;
    }
    return !1;
  } });
}
let hr = ["textarea", "input"].join(",");
function yr(t) {
  var a, e;
  return (e = (a = t == null ? void 0 : t.matches) == null ? void 0 : a.call(t, hr)) != null ? e : !1;
}
function br(t, a = (e) => e) {
  return t.slice().sort((e, l) => {
    let n = a(e), i = a(l);
    if (n === null || i === null)
      return 0;
    let V = n.compareDocumentPosition(i);
    return V & Node.DOCUMENT_POSITION_FOLLOWING ? -1 : V & Node.DOCUMENT_POSITION_PRECEDING ? 1 : 0;
  });
}
function Ve(t, a, e = !0, l = null) {
  var n;
  let i = (n = Array.isArray(t) ? t.length > 0 ? t[0].ownerDocument : document : t == null ? void 0 : t.ownerDocument) != null ? n : document, V = Array.isArray(t) ? e ? br(t) : t : st(t);
  l = l != null ? l : i.activeElement;
  let c = (() => {
    if (a & 5)
      return 1;
    if (a & 10)
      return -1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), f = (() => {
    if (a & 1)
      return 0;
    if (a & 2)
      return Math.max(0, V.indexOf(l)) - 1;
    if (a & 4)
      return Math.max(0, V.indexOf(l)) + 1;
    if (a & 8)
      return V.length - 1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), h = a & 32 ? { preventScroll: !0 } : {}, $ = 0, b = V.length, w;
  do {
    if ($ >= b || $ + b <= 0)
      return 0;
    let A = f + $;
    if (a & 16)
      A = (A + b) % b;
    else {
      if (A < 0)
        return 3;
      if (A >= b)
        return 1;
    }
    w = V[A], w == null || w.focus(h), $ += c;
  } while (w !== i.activeElement);
  return a & 6 && yr(w) && w.select(), w.hasAttribute("tabindex") || w.setAttribute("tabindex", "0"), 2;
}
function ze(t, a, e) {
  He || be((l) => {
    document.addEventListener(t, a, e), l(() => document.removeEventListener(t, a, e));
  });
}
function xr(t, a, e = de(() => !0)) {
  function l(i, V) {
    if (!e.value || i.defaultPrevented)
      return;
    let c = V(i);
    if (c === null || !c.ownerDocument.documentElement.contains(c))
      return;
    let f = function h($) {
      return typeof $ == "function" ? h($()) : Array.isArray($) || $ instanceof Set ? $ : [$];
    }(t);
    for (let h of f) {
      if (h === null)
        continue;
      let $ = h instanceof HTMLElement ? h : U(h);
      if ($ != null && $.contains(c))
        return;
    }
    return !lt(c, Ge.Loose) && c.tabIndex !== -1 && i.preventDefault(), a(i, c);
  }
  let n = ee(null);
  ze("mousedown", (i) => {
    e.value && (n.value = i.target);
  }, !0), ze("click", (i) => {
    !n.value || (l(i, () => n.value), n.value = null);
  }, !0), ze("blur", (i) => l(i, () => window.document.activeElement instanceof HTMLIFrameElement ? window.document.activeElement : null), !0);
}
var Ne = ((t) => (t[t.None = 1] = "None", t[t.Focusable = 2] = "Focusable", t[t.Hidden = 4] = "Hidden", t))(Ne || {});
let Ze = Ye({ name: "Hidden", props: { as: { type: [Object, String], default: "div" }, features: { type: Number, default: 1 } }, setup(t, { slots: a, attrs: e }) {
  return () => {
    let { features: l, ...n } = t, i = { "aria-hidden": (l & 2) === 2 ? !0 : void 0, style: { position: "fixed", top: 1, left: 1, width: 1, height: 0, padding: 0, margin: -1, overflow: "hidden", clip: "rect(0, 0, 0, 0)", whiteSpace: "nowrap", borderWidth: "0", ...(l & 4) === 4 && (l & 2) !== 2 && { display: "none" } } };
    return Te({ ourProps: i, theirProps: n, slot: {}, attrs: e, slots: a, name: "Hidden" });
  };
} });
function gr(t, a, e) {
  He || be((l) => {
    window.addEventListener(t, a, e), l(() => window.removeEventListener(t, a, e));
  });
}
var Me = ((t) => (t[t.Forwards = 0] = "Forwards", t[t.Backwards = 1] = "Backwards", t))(Me || {});
function it() {
  let t = ee(0);
  return gr("keydown", (a) => {
    a.key === "Tab" && (t.value = a.shiftKey ? 1 : 0);
  }), t;
}
function wr(t, a, e, l) {
  He || be((n) => {
    t = t != null ? t : window, t.addEventListener(a, e, l), n(() => t.removeEventListener(a, e, l));
  });
}
var kr = ((t) => (t[t.Open = 0] = "Open", t[t.Closed = 1] = "Closed", t))(kr || {});
let ut = Symbol("PopoverContext");
function Re(t) {
  let a = ue(ut, null);
  if (a === null) {
    let e = new Error(`<${t} /> is missing a parent <${mt.name} /> component.`);
    throw Error.captureStackTrace && Error.captureStackTrace(e, Re), e;
  }
  return a;
}
let dt = Symbol("PopoverGroupContext");
function ct() {
  return ue(dt, null);
}
let ft = Symbol("PopoverPanelContext");
function $r() {
  return ue(ft, null);
}
let mt = Ye({ name: "Popover", props: { as: { type: [Object, String], default: "div" } }, setup(t, { slots: a, attrs: e, expose: l }) {
  var n;
  let i = `headlessui-popover-button-${Ce()}`, V = `headlessui-popover-panel-${Ce()}`, c = ee(null);
  l({ el: c, $el: c });
  let f = ee(1), h = ee(null), $ = ee(null), b = ee(null), w = ee(null), A = de(() => Fe(c)), Z = de(() => {
    if (!U(h) || !U(w))
      return !1;
    for (let B of document.querySelectorAll("body > *"))
      if (Number(B == null ? void 0 : B.contains(U(h))) ^ Number(B == null ? void 0 : B.contains(U(w))))
        return !0;
    return !1;
  }), j = { popoverState: f, buttonId: i, panelId: V, panel: w, button: h, isPortalled: Z, beforePanelSentinel: $, afterPanelSentinel: b, togglePopover() {
    f.value = we(f.value, { [0]: 1, [1]: 0 });
  }, closePopover() {
    f.value !== 1 && (f.value = 1);
  }, close(B) {
    j.closePopover();
    let u = (() => B ? B instanceof HTMLElement ? B : B.value instanceof HTMLElement ? U(B) : U(j.button) : U(j.button))();
    u == null || u.focus();
  } };
  ie(ut, j), fr(de(() => we(f.value, { [0]: Be.Open, [1]: Be.Closed })));
  let y = { buttonId: i, panelId: V, close() {
    j.closePopover();
  } }, p = ct(), x = p == null ? void 0 : p.registerPopover;
  function Y() {
    var B, u, r, S;
    return (S = p == null ? void 0 : p.isFocusWithinPopoverGroup()) != null ? S : ((B = A.value) == null ? void 0 : B.activeElement) && (((u = U(h)) == null ? void 0 : u.contains(A.value.activeElement)) || ((r = U(w)) == null ? void 0 : r.contains(A.value.activeElement)));
  }
  return be(() => x == null ? void 0 : x(y)), wr((n = A.value) == null ? void 0 : n.defaultView, "focus", (B) => {
    var u, r;
    f.value === 0 && (Y() || !h || !w || (u = U(j.beforePanelSentinel)) != null && u.contains(B.target) || (r = U(j.afterPanelSentinel)) != null && r.contains(B.target) || j.closePopover());
  }, !0), xr([h, w], (B, u) => {
    var r;
    j.closePopover(), lt(u, Ge.Loose) || (B.preventDefault(), (r = U(h)) == null || r.focus());
  }, de(() => f.value === 0)), () => {
    let B = { open: f.value === 0, close: j.close };
    return Te({ theirProps: t, ourProps: { ref: c }, slot: B, slots: a, attrs: e, name: "Popover" });
  };
} }), jr = Ye({ name: "PopoverButton", props: { as: { type: [Object, String], default: "button" }, disabled: { type: [Boolean], default: !1 } }, inheritAttrs: !1, setup(t, { attrs: a, slots: e, expose: l }) {
  let n = Re("PopoverButton"), i = de(() => Fe(n.button));
  l({ el: n.button, $el: n.button });
  let V = ct(), c = V == null ? void 0 : V.closeOthers, f = $r(), h = f === null ? !1 : f === n.panelId, $ = ee(null), b = `headlessui-focus-sentinel-${Ce()}`;
  h || be(() => {
    n.button.value = $.value;
  });
  let w = mr(de(() => ({ as: t.as, type: a.type })), $);
  function A(p) {
    var x, Y, B, u, r;
    if (h) {
      if (n.popoverState.value === 1)
        return;
      switch (p.key) {
        case ge.Space:
        case ge.Enter:
          p.preventDefault(), (Y = (x = p.target).click) == null || Y.call(x), n.closePopover(), (B = U(n.button)) == null || B.focus();
          break;
      }
    } else
      switch (p.key) {
        case ge.Space:
        case ge.Enter:
          p.preventDefault(), p.stopPropagation(), n.popoverState.value === 1 && (c == null || c(n.buttonId)), n.togglePopover();
          break;
        case ge.Escape:
          if (n.popoverState.value !== 0)
            return c == null ? void 0 : c(n.buttonId);
          if (!U(n.button) || ((u = i.value) == null ? void 0 : u.activeElement) && !((r = U(n.button)) != null && r.contains(i.value.activeElement)))
            return;
          p.preventDefault(), p.stopPropagation(), n.closePopover();
          break;
      }
  }
  function Z(p) {
    h || p.key === ge.Space && p.preventDefault();
  }
  function j(p) {
    var x, Y;
    t.disabled || (h ? (n.closePopover(), (x = U(n.button)) == null || x.focus()) : (p.preventDefault(), p.stopPropagation(), n.popoverState.value === 1 && (c == null || c(n.buttonId)), n.togglePopover(), (Y = U(n.button)) == null || Y.focus()));
  }
  function y(p) {
    p.preventDefault(), p.stopPropagation();
  }
  return () => {
    let p = n.popoverState.value === 0, x = { open: p }, Y = h ? { ref: $, type: w.value, onKeydown: A, onClick: j } : { ref: $, id: n.buttonId, type: w.value, "aria-expanded": t.disabled ? void 0 : n.popoverState.value === 0, "aria-controls": U(n.panel) ? n.panelId : void 0, disabled: t.disabled ? !0 : void 0, onKeydown: A, onKeyup: Z, onClick: j, onMousedown: y }, B = it();
    function u() {
      let r = U(n.panel);
      if (!r)
        return;
      function S() {
        we(B.value, { [Me.Forwards]: () => Ve(r, je.First), [Me.Backwards]: () => Ve(r, je.Last) });
      }
      S();
    }
    return _e(ke, [Te({ ourProps: Y, theirProps: { ...a, ...t }, slot: x, attrs: a, slots: e, name: "PopoverButton" }), p && !h && n.isPortalled.value && _e(Ze, { id: b, features: Ne.Focusable, as: "button", type: "button", onFocus: u })]);
  };
} }), Mr = Ye({ name: "PopoverOverlay", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 } }, setup(t, { attrs: a, slots: e }) {
  let l = Re("PopoverOverlay"), n = `headlessui-popover-overlay-${Ce()}`, i = nt(), V = de(() => i !== null ? i.value === Be.Open : l.popoverState.value === 0);
  function c() {
    l.closePopover();
  }
  return () => {
    let f = { open: l.popoverState.value === 0 };
    return Te({ ourProps: { id: n, "aria-hidden": !0, onClick: c }, theirProps: t, slot: f, attrs: a, slots: e, features: Le.RenderStrategy | Le.Static, visible: V.value, name: "PopoverOverlay" });
  };
} }), Sr = Ye({ name: "PopoverPanel", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 }, focus: { type: Boolean, default: !1 } }, inheritAttrs: !1, setup(t, { attrs: a, slots: e, expose: l }) {
  let { focus: n } = t, i = Re("PopoverPanel"), V = de(() => Fe(i.panel)), c = `headlessui-focus-sentinel-before-${Ce()}`, f = `headlessui-focus-sentinel-after-${Ce()}`;
  l({ el: i.panel, $el: i.panel }), ie(ft, i.panelId), be(() => {
    var y, p;
    if (!n || i.popoverState.value !== 0 || !i.panel)
      return;
    let x = (y = V.value) == null ? void 0 : y.activeElement;
    (p = U(i.panel)) != null && p.contains(x) || Ve(U(i.panel), je.First);
  });
  let h = nt(), $ = de(() => h !== null ? h.value === Be.Open : i.popoverState.value === 0);
  function b(y) {
    var p, x;
    switch (y.key) {
      case ge.Escape:
        if (i.popoverState.value !== 0 || !U(i.panel) || V.value && !((p = U(i.panel)) != null && p.contains(V.value.activeElement)))
          return;
        y.preventDefault(), y.stopPropagation(), i.closePopover(), (x = U(i.button)) == null || x.focus();
        break;
    }
  }
  function w(y) {
    var p, x, Y, B, u;
    let r = y.relatedTarget;
    !r || !U(i.panel) || (p = U(i.panel)) != null && p.contains(r) || (i.closePopover(), (((Y = (x = U(i.beforePanelSentinel)) == null ? void 0 : x.contains) == null ? void 0 : Y.call(x, r)) || ((u = (B = U(i.afterPanelSentinel)) == null ? void 0 : B.contains) == null ? void 0 : u.call(B, r))) && r.focus({ preventScroll: !0 }));
  }
  let A = it();
  function Z() {
    let y = U(i.panel);
    if (!y)
      return;
    function p() {
      we(A.value, { [Me.Forwards]: () => {
        Ve(y, je.Next);
      }, [Me.Backwards]: () => {
        var x;
        (x = U(i.button)) == null || x.focus({ preventScroll: !0 });
      } });
    }
    p();
  }
  function j() {
    let y = U(i.panel);
    if (!y)
      return;
    function p() {
      we(A.value, { [Me.Forwards]: () => {
        var x, Y;
        let B = U(i.button), u = U(i.panel);
        if (!B)
          return;
        let r = st(), S = r.indexOf(B), R = r.slice(0, S + 1), _ = [...r.slice(S + 1), ...R];
        for (let D of _.slice())
          if (((Y = (x = D == null ? void 0 : D.id) == null ? void 0 : x.startsWith) == null ? void 0 : Y.call(x, "headlessui-focus-sentinel-")) || (u == null ? void 0 : u.contains(D))) {
            let F = _.indexOf(D);
            F !== -1 && _.splice(F, 1);
          }
        Ve(_, je.First, !1);
      }, [Me.Backwards]: () => Ve(y, je.Previous) });
    }
    p();
  }
  return () => {
    let y = { open: i.popoverState.value === 0, close: i.close }, p = { ref: i.panel, id: i.panelId, onKeydown: b, onFocusout: n && i.popoverState.value === 0 ? w : void 0, tabIndex: -1 };
    return Te({ ourProps: p, theirProps: { ...a, ...ot(t, ["focus"]) }, attrs: a, slot: y, slots: { ...e, default: (...x) => {
      var Y;
      return [_e(ke, [$.value && i.isPortalled.value && _e(Ze, { id: c, ref: i.beforePanelSentinel, features: Ne.Focusable, as: "button", type: "button", onFocus: Z }), (Y = e.default) == null ? void 0 : Y.call(e, ...x), $.value && i.isPortalled.value && _e(Ze, { id: f, ref: i.afterPanelSentinel, features: Ne.Focusable, as: "button", type: "button", onFocus: j })])];
    } }, features: Le.RenderStrategy | Le.Static, visible: $.value, name: "PopoverPanel" });
  };
} });
Ye({ name: "PopoverGroup", props: { as: { type: [Object, String], default: "div" } }, setup(t, { attrs: a, slots: e, expose: l }) {
  let n = ee(null), i = ee([]), V = de(() => Fe(n));
  l({ el: n, $el: n });
  function c(b) {
    let w = i.value.indexOf(b);
    w !== -1 && i.value.splice(w, 1);
  }
  function f(b) {
    return i.value.push(b), () => {
      c(b);
    };
  }
  function h() {
    var b;
    let w = V.value;
    if (!w)
      return !1;
    let A = w.activeElement;
    return (b = U(n)) != null && b.contains(A) ? !0 : i.value.some((Z) => {
      var j, y;
      return ((j = w.getElementById(Z.buttonId)) == null ? void 0 : j.contains(A)) || ((y = w.getElementById(Z.panelId)) == null ? void 0 : y.contains(A));
    });
  }
  function $(b) {
    for (let w of i.value)
      w.buttonId !== b && w.close();
  }
  return ie(dt, { registerPopover: f, unregisterPopover: c, isFocusWithinPopoverGroup: h, closeOthers: $ }), () => Te({ ourProps: { ref: n }, theirProps: t, slot: {}, attrs: a, slots: e, name: "PopoverGroup" });
} });
var Se = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, pt = { exports: {} };
(function(t, a) {
  (function(e, l) {
    t.exports = l();
  })(Se, function() {
    var e = 1e3, l = 6e4, n = 36e5, i = "millisecond", V = "second", c = "minute", f = "hour", h = "day", $ = "week", b = "month", w = "quarter", A = "year", Z = "date", j = "Invalid Date", y = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, p = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, x = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_") }, Y = function(C, O, M) {
      var E = String(C);
      return !E || E.length >= O ? C : "" + Array(O + 1 - E.length).join(M) + C;
    }, B = { s: Y, z: function(C) {
      var O = -C.utcOffset(), M = Math.abs(O), E = Math.floor(M / 60), k = M % 60;
      return (O <= 0 ? "+" : "-") + Y(E, 2, "0") + ":" + Y(k, 2, "0");
    }, m: function C(O, M) {
      if (O.date() < M.date())
        return -C(M, O);
      var E = 12 * (M.year() - O.year()) + (M.month() - O.month()), k = O.clone().add(E, b), N = M - k < 0, T = O.clone().add(E + (N ? -1 : 1), b);
      return +(-(E + (M - k) / (N ? k - T : T - k)) || 0);
    }, a: function(C) {
      return C < 0 ? Math.ceil(C) || 0 : Math.floor(C);
    }, p: function(C) {
      return { M: b, y: A, w: $, d: h, D: Z, h: f, m: c, s: V, ms: i, Q: w }[C] || String(C || "").toLowerCase().replace(/s$/, "");
    }, u: function(C) {
      return C === void 0;
    } }, u = "en", r = {};
    r[u] = x;
    var S = function(C) {
      return C instanceof F;
    }, R = function C(O, M, E) {
      var k;
      if (!O)
        return u;
      if (typeof O == "string") {
        var N = O.toLowerCase();
        r[N] && (k = N), M && (r[N] = M, k = N);
        var T = O.split("-");
        if (!k && T.length > 1)
          return C(T[0]);
      } else {
        var z = O.name;
        r[z] = O, k = z;
      }
      return !E && k && (u = k), k || !E && u;
    }, _ = function(C, O) {
      if (S(C))
        return C.clone();
      var M = typeof O == "object" ? O : {};
      return M.date = C, M.args = arguments, new F(M);
    }, D = B;
    D.l = R, D.i = S, D.w = function(C, O) {
      return _(C, { locale: O.$L, utc: O.$u, x: O.$x, $offset: O.$offset });
    };
    var F = function() {
      function C(M) {
        this.$L = R(M.locale, null, !0), this.parse(M);
      }
      var O = C.prototype;
      return O.parse = function(M) {
        this.$d = function(E) {
          var k = E.date, N = E.utc;
          if (k === null)
            return new Date(NaN);
          if (D.u(k))
            return new Date();
          if (k instanceof Date)
            return new Date(k);
          if (typeof k == "string" && !/Z$/i.test(k)) {
            var T = k.match(y);
            if (T) {
              var z = T[2] - 1 || 0, q = (T[7] || "0").substring(0, 3);
              return N ? new Date(Date.UTC(T[1], z, T[3] || 1, T[4] || 0, T[5] || 0, T[6] || 0, q)) : new Date(T[1], z, T[3] || 1, T[4] || 0, T[5] || 0, T[6] || 0, q);
            }
          }
          return new Date(k);
        }(M), this.$x = M.x || {}, this.init();
      }, O.init = function() {
        var M = this.$d;
        this.$y = M.getFullYear(), this.$M = M.getMonth(), this.$D = M.getDate(), this.$W = M.getDay(), this.$H = M.getHours(), this.$m = M.getMinutes(), this.$s = M.getSeconds(), this.$ms = M.getMilliseconds();
      }, O.$utils = function() {
        return D;
      }, O.isValid = function() {
        return this.$d.toString() !== j;
      }, O.isSame = function(M, E) {
        var k = _(M);
        return this.startOf(E) <= k && k <= this.endOf(E);
      }, O.isAfter = function(M, E) {
        return _(M) < this.startOf(E);
      }, O.isBefore = function(M, E) {
        return this.endOf(E) < _(M);
      }, O.$g = function(M, E, k) {
        return D.u(M) ? this[E] : this.set(k, M);
      }, O.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, O.valueOf = function() {
        return this.$d.getTime();
      }, O.startOf = function(M, E) {
        var k = this, N = !!D.u(E) || E, T = D.p(M), z = function(fe, re) {
          var le = D.w(k.$u ? Date.UTC(k.$y, re, fe) : new Date(k.$y, re, fe), k);
          return N ? le : le.endOf(h);
        }, q = function(fe, re) {
          return D.w(k.toDate()[fe].apply(k.toDate("s"), (N ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(re)), k);
        }, X = this.$W, te = this.$M, ae = this.$D, ce = "set" + (this.$u ? "UTC" : "");
        switch (T) {
          case A:
            return N ? z(1, 0) : z(31, 11);
          case b:
            return N ? z(1, te) : z(0, te + 1);
          case $:
            var he = this.$locale().weekStart || 0, me = (X < he ? X + 7 : X) - he;
            return z(N ? ae - me : ae + (6 - me), te);
          case h:
          case Z:
            return q(ce + "Hours", 0);
          case f:
            return q(ce + "Minutes", 1);
          case c:
            return q(ce + "Seconds", 2);
          case V:
            return q(ce + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, O.endOf = function(M) {
        return this.startOf(M, !1);
      }, O.$set = function(M, E) {
        var k, N = D.p(M), T = "set" + (this.$u ? "UTC" : ""), z = (k = {}, k[h] = T + "Date", k[Z] = T + "Date", k[b] = T + "Month", k[A] = T + "FullYear", k[f] = T + "Hours", k[c] = T + "Minutes", k[V] = T + "Seconds", k[i] = T + "Milliseconds", k)[N], q = N === h ? this.$D + (E - this.$W) : E;
        if (N === b || N === A) {
          var X = this.clone().set(Z, 1);
          X.$d[z](q), X.init(), this.$d = X.set(Z, Math.min(this.$D, X.daysInMonth())).$d;
        } else
          z && this.$d[z](q);
        return this.init(), this;
      }, O.set = function(M, E) {
        return this.clone().$set(M, E);
      }, O.get = function(M) {
        return this[D.p(M)]();
      }, O.add = function(M, E) {
        var k, N = this;
        M = Number(M);
        var T = D.p(E), z = function(te) {
          var ae = _(N);
          return D.w(ae.date(ae.date() + Math.round(te * M)), N);
        };
        if (T === b)
          return this.set(b, this.$M + M);
        if (T === A)
          return this.set(A, this.$y + M);
        if (T === h)
          return z(1);
        if (T === $)
          return z(7);
        var q = (k = {}, k[c] = l, k[f] = n, k[V] = e, k)[T] || 1, X = this.$d.getTime() + M * q;
        return D.w(X, this);
      }, O.subtract = function(M, E) {
        return this.add(-1 * M, E);
      }, O.format = function(M) {
        var E = this, k = this.$locale();
        if (!this.isValid())
          return k.invalidDate || j;
        var N = M || "YYYY-MM-DDTHH:mm:ssZ", T = D.z(this), z = this.$H, q = this.$m, X = this.$M, te = k.weekdays, ae = k.months, ce = function(re, le, pe, ve) {
          return re && (re[le] || re(E, N)) || pe[le].slice(0, ve);
        }, he = function(re) {
          return D.s(z % 12 || 12, re, "0");
        }, me = k.meridiem || function(re, le, pe) {
          var ve = re < 12 ? "AM" : "PM";
          return pe ? ve.toLowerCase() : ve;
        }, fe = { YY: String(this.$y).slice(-2), YYYY: this.$y, M: X + 1, MM: D.s(X + 1, 2, "0"), MMM: ce(k.monthsShort, X, ae, 3), MMMM: ce(ae, X), D: this.$D, DD: D.s(this.$D, 2, "0"), d: String(this.$W), dd: ce(k.weekdaysMin, this.$W, te, 2), ddd: ce(k.weekdaysShort, this.$W, te, 3), dddd: te[this.$W], H: String(z), HH: D.s(z, 2, "0"), h: he(1), hh: he(2), a: me(z, q, !0), A: me(z, q, !1), m: String(q), mm: D.s(q, 2, "0"), s: String(this.$s), ss: D.s(this.$s, 2, "0"), SSS: D.s(this.$ms, 3, "0"), Z: T };
        return N.replace(p, function(re, le) {
          return le || fe[re] || T.replace(":", "");
        });
      }, O.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, O.diff = function(M, E, k) {
        var N, T = D.p(E), z = _(M), q = (z.utcOffset() - this.utcOffset()) * l, X = this - z, te = D.m(this, z);
        return te = (N = {}, N[A] = te / 12, N[b] = te, N[w] = te / 3, N[$] = (X - q) / 6048e5, N[h] = (X - q) / 864e5, N[f] = X / n, N[c] = X / l, N[V] = X / e, N)[T] || X, k ? te : D.a(te);
      }, O.daysInMonth = function() {
        return this.endOf(b).$D;
      }, O.$locale = function() {
        return r[this.$L];
      }, O.locale = function(M, E) {
        if (!M)
          return this.$L;
        var k = this.clone(), N = R(M, E, !0);
        return N && (k.$L = N), k;
      }, O.clone = function() {
        return D.w(this.$d, this);
      }, O.toDate = function() {
        return new Date(this.valueOf());
      }, O.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, O.toISOString = function() {
        return this.$d.toISOString();
      }, O.toString = function() {
        return this.$d.toUTCString();
      }, C;
    }(), H = F.prototype;
    return _.prototype = H, [["$ms", i], ["$s", V], ["$m", c], ["$H", f], ["$W", h], ["$M", b], ["$y", A], ["$D", Z]].forEach(function(C) {
      H[C[1]] = function(O) {
        return this.$g(O, C[0], C[1]);
      };
    }), _.extend = function(C, O) {
      return C.$i || (C(O, F, _), C.$i = !0), _;
    }, _.locale = R, _.isDayjs = S, _.unix = function(C) {
      return _(1e3 * C);
    }, _.en = r[u], _.Ls = r, _.p = {}, _;
  });
})(pt);
const s = pt.exports;
var vt = { exports: {} };
(function(t, a) {
  (function(e, l) {
    t.exports = l();
  })(Se, function() {
    return function(e, l, n) {
      var i = l.prototype, V = function(b) {
        return b && (b.indexOf ? b : b.s);
      }, c = function(b, w, A, Z, j) {
        var y = b.name ? b : b.$locale(), p = V(y[w]), x = V(y[A]), Y = p || x.map(function(u) {
          return u.slice(0, Z);
        });
        if (!j)
          return Y;
        var B = y.weekStart;
        return Y.map(function(u, r) {
          return Y[(r + (B || 0)) % 7];
        });
      }, f = function() {
        return n.Ls[n.locale()];
      }, h = function(b, w) {
        return b.formats[w] || function(A) {
          return A.replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(Z, j, y) {
            return j || y.slice(1);
          });
        }(b.formats[w.toUpperCase()]);
      }, $ = function() {
        var b = this;
        return { months: function(w) {
          return w ? w.format("MMMM") : c(b, "months");
        }, monthsShort: function(w) {
          return w ? w.format("MMM") : c(b, "monthsShort", "months", 3);
        }, firstDayOfWeek: function() {
          return b.$locale().weekStart || 0;
        }, weekdays: function(w) {
          return w ? w.format("dddd") : c(b, "weekdays");
        }, weekdaysMin: function(w) {
          return w ? w.format("dd") : c(b, "weekdaysMin", "weekdays", 2);
        }, weekdaysShort: function(w) {
          return w ? w.format("ddd") : c(b, "weekdaysShort", "weekdays", 3);
        }, longDateFormat: function(w) {
          return h(b.$locale(), w);
        }, meridiem: this.$locale().meridiem, ordinal: this.$locale().ordinal };
      };
      i.localeData = function() {
        return $.bind(this)();
      }, n.localeData = function() {
        var b = f();
        return { firstDayOfWeek: function() {
          return b.weekStart || 0;
        }, weekdays: function() {
          return n.weekdays();
        }, weekdaysShort: function() {
          return n.weekdaysShort();
        }, weekdaysMin: function() {
          return n.weekdaysMin();
        }, months: function() {
          return n.months();
        }, monthsShort: function() {
          return n.monthsShort();
        }, longDateFormat: function(w) {
          return h(b, w);
        }, meridiem: b.meridiem, ordinal: b.ordinal };
      }, n.months = function() {
        return c(f(), "months");
      }, n.monthsShort = function() {
        return c(f(), "monthsShort", "months", 3);
      }, n.weekdays = function(b) {
        return c(f(), "weekdays", null, null, b);
      }, n.weekdaysShort = function(b) {
        return c(f(), "weekdaysShort", "weekdays", 3, b);
      }, n.weekdaysMin = function(b) {
        return c(f(), "weekdaysMin", "weekdays", 2, b);
      };
    };
  });
})(vt);
const Dr = vt.exports;
var ht = { exports: {} };
(function(t, a) {
  (function(e, l) {
    t.exports = l();
  })(Se, function() {
    var e = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" };
    return function(l, n, i) {
      var V = n.prototype, c = V.format;
      i.en.formats = e, V.format = function(f) {
        f === void 0 && (f = "YYYY-MM-DDTHH:mm:ssZ");
        var h = this.$locale().formats, $ = function(b, w) {
          return b.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(A, Z, j) {
            var y = j && j.toUpperCase();
            return Z || w[j] || e[j] || w[y].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(p, x, Y) {
              return x || Y.slice(1);
            });
          });
        }(f, h === void 0 ? {} : h);
        return c.call(this, $);
      };
    };
  });
})(ht);
const Or = ht.exports;
var yt = { exports: {} };
(function(t, a) {
  (function(e, l) {
    t.exports = l();
  })(Se, function() {
    var e = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" }, l = /(\[[^[]*\])|([-_:/.,()\s]+)|(A|a|YYYY|YY?|MM?M?M?|Do|DD?|hh?|HH?|mm?|ss?|S{1,3}|z|ZZ?)/g, n = /\d\d/, i = /\d\d?/, V = /\d*[^-_:/,()\s\d]+/, c = {}, f = function(j) {
      return (j = +j) + (j > 68 ? 1900 : 2e3);
    }, h = function(j) {
      return function(y) {
        this[j] = +y;
      };
    }, $ = [/[+-]\d\d:?(\d\d)?|Z/, function(j) {
      (this.zone || (this.zone = {})).offset = function(y) {
        if (!y || y === "Z")
          return 0;
        var p = y.match(/([+-]|\d\d)/g), x = 60 * p[1] + (+p[2] || 0);
        return x === 0 ? 0 : p[0] === "+" ? -x : x;
      }(j);
    }], b = function(j) {
      var y = c[j];
      return y && (y.indexOf ? y : y.s.concat(y.f));
    }, w = function(j, y) {
      var p, x = c.meridiem;
      if (x) {
        for (var Y = 1; Y <= 24; Y += 1)
          if (j.indexOf(x(Y, 0, y)) > -1) {
            p = Y > 12;
            break;
          }
      } else
        p = j === (y ? "pm" : "PM");
      return p;
    }, A = { A: [V, function(j) {
      this.afternoon = w(j, !1);
    }], a: [V, function(j) {
      this.afternoon = w(j, !0);
    }], S: [/\d/, function(j) {
      this.milliseconds = 100 * +j;
    }], SS: [n, function(j) {
      this.milliseconds = 10 * +j;
    }], SSS: [/\d{3}/, function(j) {
      this.milliseconds = +j;
    }], s: [i, h("seconds")], ss: [i, h("seconds")], m: [i, h("minutes")], mm: [i, h("minutes")], H: [i, h("hours")], h: [i, h("hours")], HH: [i, h("hours")], hh: [i, h("hours")], D: [i, h("day")], DD: [n, h("day")], Do: [V, function(j) {
      var y = c.ordinal, p = j.match(/\d+/);
      if (this.day = p[0], y)
        for (var x = 1; x <= 31; x += 1)
          y(x).replace(/\[|\]/g, "") === j && (this.day = x);
    }], M: [i, h("month")], MM: [n, h("month")], MMM: [V, function(j) {
      var y = b("months"), p = (b("monthsShort") || y.map(function(x) {
        return x.slice(0, 3);
      })).indexOf(j) + 1;
      if (p < 1)
        throw new Error();
      this.month = p % 12 || p;
    }], MMMM: [V, function(j) {
      var y = b("months").indexOf(j) + 1;
      if (y < 1)
        throw new Error();
      this.month = y % 12 || y;
    }], Y: [/[+-]?\d+/, h("year")], YY: [n, function(j) {
      this.year = f(j);
    }], YYYY: [/\d{4}/, h("year")], Z: $, ZZ: $ };
    function Z(j) {
      var y, p;
      y = j, p = c && c.formats;
      for (var x = (j = y.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(_, D, F) {
        var H = F && F.toUpperCase();
        return D || p[F] || e[F] || p[H].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(C, O, M) {
          return O || M.slice(1);
        });
      })).match(l), Y = x.length, B = 0; B < Y; B += 1) {
        var u = x[B], r = A[u], S = r && r[0], R = r && r[1];
        x[B] = R ? { regex: S, parser: R } : u.replace(/^\[|\]$/g, "");
      }
      return function(_) {
        for (var D = {}, F = 0, H = 0; F < Y; F += 1) {
          var C = x[F];
          if (typeof C == "string")
            H += C.length;
          else {
            var O = C.regex, M = C.parser, E = _.slice(H), k = O.exec(E)[0];
            M.call(D, k), _ = _.replace(k, "");
          }
        }
        return function(N) {
          var T = N.afternoon;
          if (T !== void 0) {
            var z = N.hours;
            T ? z < 12 && (N.hours += 12) : z === 12 && (N.hours = 0), delete N.afternoon;
          }
        }(D), D;
      };
    }
    return function(j, y, p) {
      p.p.customParseFormat = !0, j && j.parseTwoDigitYear && (f = j.parseTwoDigitYear);
      var x = y.prototype, Y = x.parse;
      x.parse = function(B) {
        var u = B.date, r = B.utc, S = B.args;
        this.$u = r;
        var R = S[1];
        if (typeof R == "string") {
          var _ = S[2] === !0, D = S[3] === !0, F = _ || D, H = S[2];
          D && (H = S[2]), c = this.$locale(), !_ && H && (c = p.Ls[H]), this.$d = function(E, k, N) {
            try {
              if (["x", "X"].indexOf(k) > -1)
                return new Date((k === "X" ? 1e3 : 1) * E);
              var T = Z(k)(E), z = T.year, q = T.month, X = T.day, te = T.hours, ae = T.minutes, ce = T.seconds, he = T.milliseconds, me = T.zone, fe = new Date(), re = X || (z || q ? 1 : fe.getDate()), le = z || fe.getFullYear(), pe = 0;
              z && !q || (pe = q > 0 ? q - 1 : fe.getMonth());
              var ve = te || 0, d = ae || 0, v = ce || 0, g = he || 0;
              return me ? new Date(Date.UTC(le, pe, re, ve, d, v, g + 60 * me.offset * 1e3)) : N ? new Date(Date.UTC(le, pe, re, ve, d, v, g)) : new Date(le, pe, re, ve, d, v, g);
            } catch {
              return new Date("");
            }
          }(u, R, r), this.init(), H && H !== !0 && (this.$L = this.locale(H).$L), F && u != this.format(R) && (this.$d = new Date("")), c = {};
        } else if (R instanceof Array)
          for (var C = R.length, O = 1; O <= C; O += 1) {
            S[1] = R[O - 1];
            var M = p.apply(this, S);
            if (M.isValid()) {
              this.$d = M.$d, this.$L = M.$L, this.init();
              break;
            }
            O === C && (this.$d = new Date(""));
          }
        else
          Y.call(this, B);
      };
    };
  });
})(yt);
const Vr = yt.exports;
var bt = { exports: {} };
(function(t, a) {
  (function(e, l) {
    t.exports = l();
  })(Se, function() {
    return function(e, l, n) {
      l.prototype.isToday = function() {
        var i = "YYYY-MM-DD", V = n();
        return this.format(i) === V.format(i);
      };
    };
  });
})(bt);
const Pr = bt.exports;
var xt = { exports: {} };
(function(t, a) {
  (function(e, l) {
    t.exports = l();
  })(Se, function() {
    return function(e, l, n) {
      l.prototype.isBetween = function(i, V, c, f) {
        var h = n(i), $ = n(V), b = (f = f || "()")[0] === "(", w = f[1] === ")";
        return (b ? this.isAfter(h, c) : !this.isBefore(h, c)) && (w ? this.isBefore($, c) : !this.isAfter($, c)) || (b ? this.isBefore(h, c) : !this.isAfter(h, c)) && (w ? this.isAfter($, c) : !this.isBefore($, c));
      };
    };
  });
})(xt);
const _r = xt.exports;
var gt = { exports: {} };
(function(t, a) {
  (function(e, l) {
    t.exports = l();
  })(Se, function() {
    var e, l, n = 1e3, i = 6e4, V = 36e5, c = 864e5, f = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, h = 31536e6, $ = 2592e6, b = /^(-|\+)?P(?:([-+]?[0-9,.]*)Y)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)W)?(?:([-+]?[0-9,.]*)D)?(?:T(?:([-+]?[0-9,.]*)H)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)S)?)?$/, w = { years: h, months: $, days: c, hours: V, minutes: i, seconds: n, milliseconds: 1, weeks: 6048e5 }, A = function(u) {
      return u instanceof B;
    }, Z = function(u, r, S) {
      return new B(u, S, r.$l);
    }, j = function(u) {
      return l.p(u) + "s";
    }, y = function(u) {
      return u < 0;
    }, p = function(u) {
      return y(u) ? Math.ceil(u) : Math.floor(u);
    }, x = function(u) {
      return Math.abs(u);
    }, Y = function(u, r) {
      return u ? y(u) ? { negative: !0, format: "" + x(u) + r } : { negative: !1, format: "" + u + r } : { negative: !1, format: "" };
    }, B = function() {
      function u(S, R, _) {
        var D = this;
        if (this.$d = {}, this.$l = _, S === void 0 && (this.$ms = 0, this.parseFromMilliseconds()), R)
          return Z(S * w[j(R)], this);
        if (typeof S == "number")
          return this.$ms = S, this.parseFromMilliseconds(), this;
        if (typeof S == "object")
          return Object.keys(S).forEach(function(C) {
            D.$d[j(C)] = S[C];
          }), this.calMilliseconds(), this;
        if (typeof S == "string") {
          var F = S.match(b);
          if (F) {
            var H = F.slice(2).map(function(C) {
              return C != null ? Number(C) : 0;
            });
            return this.$d.years = H[0], this.$d.months = H[1], this.$d.weeks = H[2], this.$d.days = H[3], this.$d.hours = H[4], this.$d.minutes = H[5], this.$d.seconds = H[6], this.calMilliseconds(), this;
          }
        }
        return this;
      }
      var r = u.prototype;
      return r.calMilliseconds = function() {
        var S = this;
        this.$ms = Object.keys(this.$d).reduce(function(R, _) {
          return R + (S.$d[_] || 0) * w[_];
        }, 0);
      }, r.parseFromMilliseconds = function() {
        var S = this.$ms;
        this.$d.years = p(S / h), S %= h, this.$d.months = p(S / $), S %= $, this.$d.days = p(S / c), S %= c, this.$d.hours = p(S / V), S %= V, this.$d.minutes = p(S / i), S %= i, this.$d.seconds = p(S / n), S %= n, this.$d.milliseconds = S;
      }, r.toISOString = function() {
        var S = Y(this.$d.years, "Y"), R = Y(this.$d.months, "M"), _ = +this.$d.days || 0;
        this.$d.weeks && (_ += 7 * this.$d.weeks);
        var D = Y(_, "D"), F = Y(this.$d.hours, "H"), H = Y(this.$d.minutes, "M"), C = this.$d.seconds || 0;
        this.$d.milliseconds && (C += this.$d.milliseconds / 1e3);
        var O = Y(C, "S"), M = S.negative || R.negative || D.negative || F.negative || H.negative || O.negative, E = F.format || H.format || O.format ? "T" : "", k = (M ? "-" : "") + "P" + S.format + R.format + D.format + E + F.format + H.format + O.format;
        return k === "P" || k === "-P" ? "P0D" : k;
      }, r.toJSON = function() {
        return this.toISOString();
      }, r.format = function(S) {
        var R = S || "YYYY-MM-DDTHH:mm:ss", _ = { Y: this.$d.years, YY: l.s(this.$d.years, 2, "0"), YYYY: l.s(this.$d.years, 4, "0"), M: this.$d.months, MM: l.s(this.$d.months, 2, "0"), D: this.$d.days, DD: l.s(this.$d.days, 2, "0"), H: this.$d.hours, HH: l.s(this.$d.hours, 2, "0"), m: this.$d.minutes, mm: l.s(this.$d.minutes, 2, "0"), s: this.$d.seconds, ss: l.s(this.$d.seconds, 2, "0"), SSS: l.s(this.$d.milliseconds, 3, "0") };
        return R.replace(f, function(D, F) {
          return F || String(_[D]);
        });
      }, r.as = function(S) {
        return this.$ms / w[j(S)];
      }, r.get = function(S) {
        var R = this.$ms, _ = j(S);
        return _ === "milliseconds" ? R %= 1e3 : R = _ === "weeks" ? p(R / w[_]) : this.$d[_], R === 0 ? 0 : R;
      }, r.add = function(S, R, _) {
        var D;
        return D = R ? S * w[j(R)] : A(S) ? S.$ms : Z(S, this).$ms, Z(this.$ms + D * (_ ? -1 : 1), this);
      }, r.subtract = function(S, R) {
        return this.add(S, R, !0);
      }, r.locale = function(S) {
        var R = this.clone();
        return R.$l = S, R;
      }, r.clone = function() {
        return Z(this.$ms, this);
      }, r.humanize = function(S) {
        return e().add(this.$ms, "ms").locale(this.$l).fromNow(!S);
      }, r.milliseconds = function() {
        return this.get("milliseconds");
      }, r.asMilliseconds = function() {
        return this.as("milliseconds");
      }, r.seconds = function() {
        return this.get("seconds");
      }, r.asSeconds = function() {
        return this.as("seconds");
      }, r.minutes = function() {
        return this.get("minutes");
      }, r.asMinutes = function() {
        return this.as("minutes");
      }, r.hours = function() {
        return this.get("hours");
      }, r.asHours = function() {
        return this.as("hours");
      }, r.days = function() {
        return this.get("days");
      }, r.asDays = function() {
        return this.as("days");
      }, r.weeks = function() {
        return this.get("weeks");
      }, r.asWeeks = function() {
        return this.as("weeks");
      }, r.months = function() {
        return this.get("months");
      }, r.asMonths = function() {
        return this.as("months");
      }, r.years = function() {
        return this.get("years");
      }, r.asYears = function() {
        return this.as("years");
      }, u;
    }();
    return function(u, r, S) {
      e = S, l = S().$utils(), S.duration = function(D, F) {
        var H = S.locale();
        return Z(D, { $l: H }, F);
      }, S.isDuration = A;
      var R = r.prototype.add, _ = r.prototype.subtract;
      r.prototype.add = function(D, F) {
        return A(D) && (D = D.asMilliseconds()), R.bind(this)(D, F);
      }, r.prototype.subtract = function(D, F) {
        return A(D) && (D = D.asMilliseconds()), _.bind(this)(D, F);
      };
    };
  });
})(gt);
const Cr = gt.exports;
function Yr() {
  const t = (c) => {
    const f = [];
    for (let h = 0; h <= c.date(0).day(); h++)
      f.push(c.date(0).subtract(h, "day"));
    return f.sort((h, $) => h.date() - $.date());
  };
  return {
    usePreviousDate: t,
    useCurrentDate: (c) => Array.from(
      {
        length: c.daysInMonth()
      },
      (f, h) => c.date(h + 1)
    ),
    useNextDate: (c) => {
      const f = [];
      for (let h = 1; h <= 42 - (t(c).length + c.daysInMonth()); h++)
        f.push(c.date(h).month(c.month()).add(1, "month"));
      return f;
    },
    useDisableDate: (c, { disableDate: f }) => typeof f == "function" ? f(c.toDate()) : !1,
    useBetweenRange: (c, { previous: f, next: h }) => {
      let $;
      return f.isAfter(h, "date") ? $ = "(]" : $ = "[)", !!(c.isBetween(f, h, "date", $) && !c.off);
    },
    useToValueFromString: (c, { formatter: f }) => c.format(f.date),
    useToValueFromArray: ({ previous: c, next: f }, { formatter: h, separator: $ }) => `${c.format(h.date)}${$}${f.format(h.date)}`
  };
}
function Tr() {
  return {
    useVisibleViewport: (a) => {
      if (a) {
        const { right: e } = a.getBoundingClientRect(), l = window.innerWidth || document.documentElement.clientWidth;
        return e > l;
      } else
        return null;
    }
  };
}
const Ar = ["placeholder"], Lr = { class: "absolute inset-y-0 right-0 inline-flex items-center rounded-md overflow-hidden" }, Br = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Er = {
  key: 0,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M6 18L18 6M6 6l12 12"
}, Fr = {
  key: 1,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
}, Nr = { class: "flex flex-wrap lg:flex-nowrap" }, Hr = { class: "relative flex flex-wrap sm:flex-nowrap p-1" }, Rr = {
  key: 0,
  class: "hidden h-full absolute inset-0 sm:flex justify-center items-center"
}, Ir = /* @__PURE__ */ L("div", { class: "h-full border-r border-black/[.1] dark:border-vtd-secondary-700/[1]" }, null, -1), Ur = [
  Ir
], zr = { class: "px-0.5 sm:px-2" }, Wr = {
  key: 1,
  class: "relative w-full sm:w-80 overflow-hidden mt-3 sm:mt-0 sm:ml-2"
}, Zr = { class: "px-0.5 sm:px-2" }, Gr = { key: 0 }, Kr = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, Jr = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, qr = ["disabled", "onClick", "textContent"], Xr = ["onClick", "textContent"], Qr = {
  key: 1,
  class: "sm:hidden"
}, eo = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, to = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, ro = ["onClick", "textContent"], oo = {
  __name: "VueTailwindDatePicker",
  props: {
    overlay: Boolean,
    asSingle: Boolean,
    useRange: Boolean,
    placeholder: {
      type: [Boolean, String],
      default: !1
    },
    i18n: {
      type: String,
      default: "en"
    },
    inputClasses: {
      type: String,
      default: ""
    },
    disableInRange: {
      type: Boolean,
      default: !0
    },
    disableDate: {
      type: [Boolean, Array, Function],
      default: !1
    },
    autoApply: {
      type: Boolean,
      default: !0
    },
    shortcuts: {
      type: [Boolean, Function],
      default: !0
    },
    separator: {
      type: String,
      default: " ~ "
    },
    formatter: {
      type: Object,
      default: () => ({
        date: "YYYY-MM-DD HH:mm:ss",
        month: "MMM"
      })
    },
    modelValue: {
      type: [Array, Object, String],
      default: () => []
    },
    startFrom: {
      type: [Object, String],
      default: () => new Date()
    },
    options: {
      type: Object,
      default: () => ({
        shortcuts: {
          today: "Today",
          yesterday: "Yesterday",
          past: (t) => `Last ${t} Days`,
          currentMonth: "This Month",
          pastMonth: "Last Month"
        },
        footer: {
          apply: "Apply",
          cancel: "Cancel"
        }
      })
    }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: a }) {
    const e = t, {
      useCurrentDate: l,
      useDisableDate: n,
      useBetweenRange: i,
      useNextDate: V,
      usePreviousDate: c,
      useToValueFromArray: f,
      useToValueFromString: h
    } = Yr(), { useVisibleViewport: $ } = Tr();
    s.extend(Dr), s.extend(Or), s.extend(Vr), s.extend(Pr), s.extend(_r), s.extend(Cr);
    const b = ee(null), w = ee(null), A = ee(null), Z = ee(""), j = ee(null), y = ee(""), p = ee([]), x = ee([]), Y = ee(null), B = ee(null), u = Mt({
      previous: {
        calendar: !0,
        month: !1,
        year: !1
      },
      next: {
        calendar: !0,
        month: !1,
        year: !1
      }
    }), r = ee({
      previous: s(),
      next: s().add(1, "month"),
      year: {
        previous: s().year(),
        next: s().year()
      },
      weeks: s.weekdaysShort(),
      months: e.formatter.month === "MMM" ? s.monthsShort() : s.months()
    }), S = de(() => r.value.weeks), R = de(() => r.value.months), _ = de(() => {
      const { previous: d, next: v, year: g } = G(r);
      return {
        previous: {
          date: () => c(d).concat(l(d)).concat(V(d)).map((o) => (o.today = o.isToday(), o.active = d.month() === o.month(), o.off = d.month() !== o.month(), o.sunday = o.day() === 0, o.disabled = n(o, e) && !C(o), o.inRange = () => {
            if (e.asSingle && !e.useRange)
              return d.month() !== o.month();
          }, o.hovered = () => H() && p.value.length > 1 ? (o.isBetween(p.value[0], p.value[1], "date", "()") || o.isBetween(p.value[1], p.value[0], "date", "()")) && d.month() === o.month() : !1, o.duration = () => !1, o)),
          month: d && d.format(e.formatter.month),
          year: d && d.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (o, m) => g.previous + m
          ),
          onPrevious: () => {
            r.value.previous = d.subtract(1, "month");
          },
          onNext: () => {
            r.value.previous = d.add(1, "month"), d.diff(v, "month") === -1 && (r.value.next = v.add(1, "month"));
          },
          onPreviousYear: () => {
            r.value.year.previous = r.value.year.previous - 12;
          },
          onNextYear: () => {
            r.value.year.previous = r.value.year.previous + 12;
          },
          openMonth: () => {
            u.previous.month = !u.previous.month, u.previous.year = !1, u.previous.calendar = !u.previous.month;
          },
          setMount: (o) => {
            r.value.previous = d.month(o), u.previous.month = !u.previous.month, u.previous.year = !1, u.previous.calendar = !u.previous.month, Ae(() => {
              (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month")), r.value.year.next = r.value.next.year();
            });
          },
          openYear: () => {
            u.previous.year = !u.previous.year, u.previous.month = !1, u.previous.calendar = !u.previous.year;
          },
          setYear: (o, m) => {
            m || (r.value.previous = d.year(o), u.previous.year = !u.previous.year, u.previous.calendar = !u.previous.year, Ae(() => {
              (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month")), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year();
            }));
          }
        },
        next: {
          date: () => c(v).concat(l(v)).concat(V(v)).map((o) => (o.today = o.isToday(), o.active = v.month() === o.month(), o.off = v.month() !== o.month(), o.sunday = o.day() === 0, o.disabled = n(o, e) && !C(o), o.inRange = () => {
            if (e.asSingle && !e.useRange)
              return v.month() !== o.month();
          }, o.hovered = () => p.value.length > 1 ? (o.isBetween(p.value[0], p.value[1], "date", "()") || o.isBetween(p.value[1], p.value[0], "date", "()")) && v.month() === o.month() : !1, o.duration = () => !1, o)),
          month: v && v.format(e.formatter.month),
          year: v && v.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (o, m) => g.next + m
          ),
          onPrevious: () => {
            r.value.next = v.subtract(1, "month"), v.diff(d, "month") === 1 && (r.value.previous = d.subtract(1, "month"));
          },
          onNext: () => {
            r.value.next = v.add(1, "month");
          },
          onPreviousYear: () => {
            r.value.year.next = r.value.year.next - 12;
          },
          onNextYear: () => {
            r.value.year.next = r.value.year.next + 12;
          },
          openMonth: () => {
            u.next.month = !u.next.month, u.next.year = !1, u.next.calendar = !u.next.month;
          },
          setMount: (o) => {
            r.value.next = v.month(o), u.next.month = !u.next.month, u.next.year = !1, u.next.calendar = !u.next.month, Ae(() => {
              (r.value.previous.isSame(r.value.next, "month") || r.value.previous.isAfter(r.value.next)) && (r.value.previous = r.value.next.subtract(1, "month")), r.value.year.previous = r.value.previous.year();
            });
          },
          openYear: () => {
            u.next.year = !u.next.year, u.next.month = !1, u.next.calendar = !u.next.year;
          },
          setYear: (o, m) => {
            m && (r.value.next = v.year(o), u.next.year = !u.next.year, u.next.month = !1, u.next.calendar = !u.next.year, Ae(() => {
              (r.value.previous.isSame(r.value.next, "month") || r.value.previous.isAfter(r.value.next)) && (r.value.previous = r.value.next.subtract(1, "month")), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year();
            }));
          }
        }
      };
    }), D = () => Array.isArray(e.modelValue), F = () => typeof e.modelValue == "object", H = () => !e.useRange && !e.asSingle ? !0 : !e.useRange && e.asSingle ? !1 : e.useRange && !e.asSingle ? !0 : !!(e.useRange && e.asSingle), C = (d) => {
      if (e.disableInRange || y.value === "")
        return !1;
      let v, g;
      if (D()) {
        const [o, m] = e.modelValue;
        v = o, g = m;
      } else if (F()) {
        if (e.modelValue) {
          const [o, m] = Object.values(e.modelValue);
          v = o, g = m;
        }
      } else {
        const [o, m] = e.modelValue.split(e.separator);
        v = o, g = m;
      }
      return d.isBetween(s(v, e.formatter.date, !0), s(g, e.formatter.date, !0), "date", "[]");
    }, O = () => {
      Y.value = null, B.value = null, p.value = [], j.value = null;
    }, M = () => {
      if (y.value = "", D())
        a("update:modelValue", []);
      else if (F()) {
        const d = {}, [v, g] = Object.keys(e.modelValue);
        d[v] = "", d[g] = "", a("update:modelValue", d);
      } else
        a("update:modelValue", "");
      x.value = [], w.value && w.value.focus();
    }, E = () => {
      if (H()) {
        const [d, v] = y.value.split(e.separator), [g, o] = [s(d, e.formatter.date, !0), s(v, e.formatter.date, !0)];
        if (g.isValid() && o.isValid())
          if (k(g), k(o), D())
            a("update:modelValue", [d, v]);
          else if (F()) {
            const m = {}, [P, W] = Object.keys(e.modelValue);
            m[P] = d, m[W] = v, a("update:modelValue", m);
          } else
            a(
              "update:modelValue",
              f(
                {
                  previous: g,
                  next: o
                },
                e
              )
            );
      } else {
        const d = s(y.value, e.formatter.date, !0);
        if (d.isValid())
          if (k(d), D())
            a("update:modelValue", [y.value]);
          else if (F()) {
            const v = {}, [g] = Object.keys(e.modelValue);
            v[g] = y.value, a("update:modelValue", v);
          } else
            a("update:modelValue", y.value);
      }
    }, k = (d, v, g) => {
      if (H())
        if (Y.value)
          if (B.value = d, e.autoApply) {
            d.isBefore(Y.value) ? y.value = f(
              {
                previous: d,
                next: Y.value
              },
              e
            ) : y.value = f(
              {
                previous: Y.value,
                next: d
              },
              e
            );
            const [o, m] = y.value.split(e.separator);
            if (D())
              a("update:modelValue", [
                s(o, e.formatter.date, !0).format(e.formatter.date),
                s(m, e.formatter.date, !0).format(e.formatter.date)
              ]);
            else if (F()) {
              const P = {}, [W, oe] = Object.keys(e.modelValue);
              P[W] = o, P[oe] = m, a("update:modelValue", P);
            } else
              a(
                "update:modelValue",
                f(
                  {
                    previous: s(o, e.formatter.date, !0),
                    next: s(m, e.formatter.date, !0)
                  },
                  e
                )
              );
            g(), x.value = [], s(o, e.formatter.date, !0).isSame(s(m, e.formatter.date, !0), "month") || (r.value.previous = s(o, e.formatter.date, !0), r.value.next = s(m, e.formatter.date, !0)), O();
          } else {
            Y.value.isAfter(d, "month") ? x.value = [d, Y.value] : x.value = [Y.value, d];
            const [o, m] = x.value;
            o.isSame(m, "month") || (r.value.previous = o, r.value.next = m), O();
          }
        else
          x.value = [], Y.value = d, j.value = d, p.value.push(d), x.value.push(d), v ? (r.value.next = d, r.value.previous.isSame(d, "month") && (r.value.next = d.add(1, "month"))) : (r.value.previous = d, r.value.next.isSame(d, "month") && (r.value.previous = r.value.next, r.value.next = d.add(1, "month")));
      else if (e.autoApply) {
        if (y.value = h(d, e), D())
          a("update:modelValue", [y.value]);
        else if (F()) {
          const o = {}, [m] = Object.keys(e.modelValue);
          o[m] = y.value, a("update:modelValue", o);
        } else
          a("update:modelValue", y.value);
        g(), x.value = [], O();
      } else
        x.value = [d], O();
    }, N = (d) => {
      if (x.value.length < 1)
        return !1;
      let v;
      if (H()) {
        const [g, o] = x.value;
        o.isBefore(g) ? v = f(
          {
            previous: o,
            next: g
          },
          e
        ) : v = f(
          {
            previous: g,
            next: o
          },
          e
        );
      } else {
        const [g] = x.value;
        v = g;
      }
      if (H()) {
        const [g, o] = v.split(e.separator);
        if (D())
          a("update:modelValue", [
            s(g, e.formatter.date, !0).format(e.formatter.date),
            s(o, e.formatter.date, !0).format(e.formatter.date)
          ]);
        else if (F()) {
          const m = {}, [P, W] = Object.keys(e.modelValue);
          m[P] = g, m[W] = o, a("update:modelValue", m);
        } else
          a(
            "update:modelValue",
            f(
              {
                previous: s(g, e.formatter.date, !0),
                next: s(o, e.formatter.date, !0)
              },
              e
            )
          );
        y.value = v;
      } else if (y.value = v.format(e.formatter.date), D())
        a("update:modelValue", [y.value]);
      else if (F()) {
        const g = {}, [o] = Object.keys(e.modelValue);
        g[o] = y.value, a("update:modelValue", g);
      } else
        a("update:modelValue", y.value);
      d();
    }, T = (d) => {
      if (!H())
        return !1;
      if (Y.value)
        p.value = [Y.value, d];
      else
        return p.value = [], !1;
    }, z = (d) => {
      if (Y.value && e.autoApply)
        return !1;
      let v, g;
      if (p.value.length > 1) {
        const [o, m] = p.value;
        v = s(o, e.formatter.date, !0), g = s(m, e.formatter.date, !0);
      } else if (D())
        if (e.autoApply) {
          const [o, m] = e.modelValue;
          v = o && s(o, e.formatter.date, !0), g = m && s(m, e.formatter.date, !0);
        } else {
          const [o, m] = x.value;
          v = s(o, e.formatter.date, !0), g = s(m, e.formatter.date, !0);
        }
      else if (F())
        if (e.autoApply) {
          if (e.modelValue) {
            const [o, m] = Object.values(e.modelValue);
            v = o && s(o, e.formatter.date, !0), g = m && s(m, e.formatter.date, !0);
          }
        } else {
          const [o, m] = x.value;
          v = s(o, e.formatter.date, !0), g = s(m, e.formatter.date, !0);
        }
      else if (e.autoApply) {
        const [o, m] = e.modelValue ? e.modelValue.split(e.separator) : [!1, !1];
        v = o && s(o, e.formatter.date, !0), g = m && s(m, e.formatter.date, !0);
      } else {
        const [o, m] = x.value;
        v = s(o, e.formatter.date, !0), g = s(m, e.formatter.date, !0);
      }
      return v && g ? i(d, {
        previous: v,
        next: g
      }) : !1;
    }, q = (d) => {
      const { today: v, active: g, off: o, disabled: m } = d;
      let P, W, oe;
      if (H())
        if (D())
          if (j.value) {
            const [I, Q] = p.value;
            W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
          } else if (e.autoApply) {
            const [I, Q] = e.modelValue;
            W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
          } else {
            const [I, Q] = x.value;
            W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
          }
        else if (F())
          if (j.value) {
            const [I, Q] = p.value;
            W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
          } else if (e.autoApply) {
            const [I, Q] = e.modelValue ? Object.values(e.modelValue) : [!1, !1];
            W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
          } else {
            const [I, Q] = x.value;
            W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
          }
        else if (j.value) {
          const [I, Q] = p.value;
          W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
        } else if (e.autoApply) {
          const [I, Q] = e.modelValue ? e.modelValue.split(e.separator) : [!1, !1];
          W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
        } else {
          const [I, Q] = x.value;
          W = I && s(I, e.formatter.date, !0), oe = Q && s(Q, e.formatter.date, !0);
        }
      else if (D())
        if (e.autoApply) {
          if (e.modelValue.length > 0) {
            const [I] = e.modelValue;
            W = s(I, e.formatter.date, !0);
          }
        } else {
          const [I] = x.value;
          W = I && s(I, e.formatter.date, !0);
        }
      else if (F())
        if (e.autoApply) {
          if (e.modelValue) {
            const [I] = Object.values(e.modelValue);
            W = s(I, e.formatter.date, !0);
          }
        } else {
          const [I] = x.value;
          W = I && s(I, e.formatter.date, !0);
        }
      else if (e.autoApply) {
        if (e.modelValue) {
          const [I] = e.modelValue.split(e.separator);
          W = s(I, e.formatter.date, !0);
        }
      } else {
        const [I] = x.value;
        W = I && s(I, e.formatter.date, !0);
      }
      return g && (P = v ? "text-vtd-primary-500 font-semibold dark:text-vtd-primary-400 rounded-full" : m ? "text-vtd-secondary-600 font-normal disabled:text-vtd-secondary-500 disabled:cursor-not-allowed rounded-full" : d.isBetween(W, oe, "date", "()") ? "text-vtd-secondary-700 font-medium dark:text-vtd-secondary-100 rounded-full" : "text-vtd-secondary-600 font-medium dark:text-vtd-secondary-200 rounded-full"), o && (P = "text-vtd-secondary-400 font-light disabled:cursor-not-allowed"), W && oe && !o ? (d.isSame(W, "date") && (P = oe.isAfter(W, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed", W.isSame(oe, "date") && (P = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed")), d.isSame(oe, "date") && (P = oe.isAfter(W, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed", W.isSame(oe, "date") && (P = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"))) : W && d.isSame(W, "date") && !o && (P = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"), P;
    }, X = (d) => {
      let v, g, o;
      if (v = "", !H())
        return v;
      if (D())
        if (p.value.length > 1) {
          const [m, P] = p.value;
          g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
        } else if (e.autoApply) {
          const [m, P] = e.modelValue;
          g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
        } else {
          const [m, P] = x.value;
          g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
        }
      else if (F())
        if (p.value.length > 1) {
          const [m, P] = p.value;
          g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
        } else if (e.autoApply) {
          if (e.modelValue) {
            const [m, P] = Object.values(e.modelValue);
            g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
          }
        } else {
          const [m, P] = x.value;
          g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
        }
      else if (p.value.length > 1) {
        const [m, P] = p.value;
        g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
      } else if (e.autoApply) {
        const [m, P] = e.modelValue ? e.modelValue.split(e.separator) : [!1, !1];
        g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
      } else {
        const [m, P] = x.value;
        g = m && s(m, e.formatter.date, !0), o = P && s(P, e.formatter.date, !0);
      }
      return g && o && (d.isSame(g, "date") ? (o.isBefore(g) && (v += " rounded-r-full inset-0"), g.isBefore(o) && (v += " rounded-l-full inset-0")) : d.isSame(o, "date") ? (o.isBefore(g) && (v += " rounded-l-full inset-0"), g.isBefore(o) && (v += " rounded-r-full inset-0")) : v += " inset-0"), v;
    }, te = (d, v) => {
      r.value.previous = s(d, e.formatter.date, !0), r.value.next = s(v, e.formatter.date, !0), (s.duration(r.value.next.diff(r.value.previous)).$d.months === 2 || s.duration(r.value.next.diff(r.value.previous)).$d.months === 1 && s.duration(r.value.next.diff(r.value.previous)).$d.days === 7) && (r.value.next = r.value.next.subtract(1, "month")), (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month"));
    }, ae = (d, v) => {
      if (H())
        if (e.autoApply) {
          if (D())
            a("update:modelValue", [d, v]);
          else if (F()) {
            const g = {}, [o, m] = Object.keys(e.modelValue);
            g[o] = d, g[m] = v, a("update:modelValue", g);
          } else
            a(
              "update:modelValue",
              f(
                {
                  previous: s(d, e.formatter.date, !0),
                  next: s(v, e.formatter.date, !0)
                },
                e
              )
            );
          y.value = `${d}${e.separator}${v}`;
        } else
          x.value = [s(d, e.formatter.date, !0), s(v, e.formatter.date, !0)];
      else if (e.autoApply) {
        if (D())
          a("update:modelValue", [d]);
        else if (F()) {
          const g = {}, [o] = Object.keys(e.modelValue);
          g[o] = d, a("update:modelValue", g);
        } else
          a("update:modelValue", d);
        y.value = d, close();
      } else
        x.value = [s(d, e.formatter.date, !0), s(v, e.formatter.date, !0)];
      te(d, v);
    }, ce = (d) => {
      const v = s().format(e.formatter.date), g = s().format(e.formatter.date);
      ae(v, g), d();
    }, he = (d) => {
      const v = s().subtract(1, "day").format(e.formatter.date), g = s().subtract(1, "day").format(e.formatter.date);
      ae(v, g), d();
    }, me = (d, v) => {
      const g = s().subtract(d - 1, "day").format(e.formatter.date), o = s().format(e.formatter.date);
      ae(g, o), v();
    }, fe = (d) => {
      const v = s().date(1).format(e.formatter.date), g = s().date(s().daysInMonth()).format(e.formatter.date);
      ae(v, g), d();
    }, re = (d) => {
      const v = s().date(1).subtract(1, "month").format(e.formatter.date), g = s().date(0).format(e.formatter.date);
      ae(v, g), d();
    }, le = (d, v) => {
      let g, o;
      const [m, P] = d.atClick();
      g = s(m).format(e.formatter.date), o = s(P).format(e.formatter.date), ae(g, o), v();
    };
    St(
      () => x.value,
      (d) => {
        d.length > 0 && (u.previous.calendar = !0, u.previous.month = !1, u.previous.year = !1, u.next.calendar = !0, u.next.month = !1, u.next.year = !1);
      }
    ), be(() => {
      e.placeholder ? Z.value = e.placeholder : H() ? Z.value = `${e.formatter.date}${e.separator}${e.formatter.date}` : Z.value = e.formatter.date;
    }), be(() => {
      const d = e.i18n;
      Ae(() => {
        const v = /* @__PURE__ */ Object.assign({ "./locale/af.js": () => import("./af.d45f0baf.mjs"), "./locale/am.js": () => import("./am.ee47f33d.mjs"), "./locale/ar-dz.js": () => import("./ar-dz.a7ea2e3d.mjs"), "./locale/ar-iq.js": () => import("./ar-iq.e35f63ce.mjs"), "./locale/ar-kw.js": () => import("./ar-kw.54002d33.mjs"), "./locale/ar-ly.js": () => import("./ar-ly.05d1b380.mjs"), "./locale/ar-ma.js": () => import("./ar-ma.2ff2aa3a.mjs"), "./locale/ar-sa.js": () => import("./ar-sa.85e93f10.mjs"), "./locale/ar-tn.js": () => import("./ar-tn.9f8c3688.mjs"), "./locale/ar.js": () => import("./ar.2cdeaca9.mjs"), "./locale/az.js": () => import("./az.469e4865.mjs"), "./locale/be.js": () => import("./be.e5e0de3d.mjs"), "./locale/bg.js": () => import("./bg.c0e75e65.mjs"), "./locale/bi.js": () => import("./bi.bd903cc6.mjs"), "./locale/bm.js": () => import("./bm.54e6e525.mjs"), "./locale/bn-bd.js": () => import("./bn-bd.3314f0e8.mjs"), "./locale/bn.js": () => import("./bn.b5046387.mjs"), "./locale/bo.js": () => import("./bo.75ec34f6.mjs"), "./locale/br.js": () => import("./br.0721b24b.mjs"), "./locale/bs.js": () => import("./bs.67503e2c.mjs"), "./locale/ca.js": () => import("./ca.2a9e059b.mjs"), "./locale/cs.js": () => import("./cs.e8b7543c.mjs"), "./locale/cv.js": () => import("./cv.825ae921.mjs"), "./locale/cy.js": () => import("./cy.664e2cba.mjs"), "./locale/da.js": () => import("./da.7195456f.mjs"), "./locale/de-at.js": () => import("./de-at.6015e5f4.mjs"), "./locale/de-ch.js": () => import("./de-ch.1a10900a.mjs"), "./locale/de.js": () => import("./de.6da03a45.mjs"), "./locale/dv.js": () => import("./dv.eba6a085.mjs"), "./locale/el.js": () => import("./el.83e00681.mjs"), "./locale/en-au.js": () => import("./en-au.023dc4bf.mjs"), "./locale/en-ca.js": () => import("./en-ca.459383dd.mjs"), "./locale/en-gb.js": () => import("./en-gb.71aad5b5.mjs"), "./locale/en-ie.js": () => import("./en-ie.137e6895.mjs"), "./locale/en-il.js": () => import("./en-il.860f1474.mjs"), "./locale/en-in.js": () => import("./en-in.cc6ca4ea.mjs"), "./locale/en-nz.js": () => import("./en-nz.64023b6f.mjs"), "./locale/en-sg.js": () => import("./en-sg.18056df6.mjs"), "./locale/en-tt.js": () => import("./en-tt.6f449368.mjs"), "./locale/en.js": () => import("./en.c83fabfa.mjs"), "./locale/eo.js": () => import("./eo.e5339261.mjs"), "./locale/es-do.js": () => import("./es-do.475484b3.mjs"), "./locale/es-mx.js": () => import("./es-mx.627a4d95.mjs"), "./locale/es-pr.js": () => import("./es-pr.1fd7e4e0.mjs"), "./locale/es-us.js": () => import("./es-us.0f4ee2d6.mjs"), "./locale/es.js": () => import("./es.37a2456a.mjs"), "./locale/et.js": () => import("./et.1403d92e.mjs"), "./locale/eu.js": () => import("./eu.c707ef35.mjs"), "./locale/fa.js": () => import("./fa.4e1f8777.mjs"), "./locale/fi.js": () => import("./fi.34767432.mjs"), "./locale/fo.js": () => import("./fo.13771a87.mjs"), "./locale/fr-ca.js": () => import("./fr-ca.ff618683.mjs"), "./locale/fr-ch.js": () => import("./fr-ch.32ded9e9.mjs"), "./locale/fr.js": () => import("./fr.d4f3dc53.mjs"), "./locale/fy.js": () => import("./fy.bd36c24c.mjs"), "./locale/ga.js": () => import("./ga.77b35dbd.mjs"), "./locale/gd.js": () => import("./gd.7c937f0d.mjs"), "./locale/gl.js": () => import("./gl.0c806545.mjs"), "./locale/gom-latn.js": () => import("./gom-latn.9469efc9.mjs"), "./locale/gu.js": () => import("./gu.58a8049d.mjs"), "./locale/he.js": () => import("./he.9d17b1ca.mjs"), "./locale/hi.js": () => import("./hi.21b0611c.mjs"), "./locale/hr.js": () => import("./hr.8bde1ff8.mjs"), "./locale/ht.js": () => import("./ht.669da1f6.mjs"), "./locale/hu.js": () => import("./hu.2ff7fb67.mjs"), "./locale/hy-am.js": () => import("./hy-am.04af3a9d.mjs"), "./locale/id.js": () => import("./id.8b963e95.mjs"), "./locale/is.js": () => import("./is.f9287e6e.mjs"), "./locale/it-ch.js": () => import("./it-ch.208ad1c6.mjs"), "./locale/it.js": () => import("./it.db20a0ad.mjs"), "./locale/ja.js": () => import("./ja.b0d3b12e.mjs"), "./locale/jv.js": () => import("./jv.54ad3888.mjs"), "./locale/ka.js": () => import("./ka.66609dab.mjs"), "./locale/kk.js": () => import("./kk.02805f2c.mjs"), "./locale/km.js": () => import("./km.e24cf400.mjs"), "./locale/kn.js": () => import("./kn.b6ca8ff7.mjs"), "./locale/ko.js": () => import("./ko.9eaed208.mjs"), "./locale/ku.js": () => import("./ku.a0eadcd4.mjs"), "./locale/ky.js": () => import("./ky.989f3453.mjs"), "./locale/lb.js": () => import("./lb.62c38266.mjs"), "./locale/lo.js": () => import("./lo.e2b48d30.mjs"), "./locale/lt.js": () => import("./lt.7c652c6c.mjs"), "./locale/lv.js": () => import("./lv.f44a13b0.mjs"), "./locale/me.js": () => import("./me.a227b562.mjs"), "./locale/mi.js": () => import("./mi.b5c535ca.mjs"), "./locale/mk.js": () => import("./mk.893e84ac.mjs"), "./locale/ml.js": () => import("./ml.74b82e97.mjs"), "./locale/mn.js": () => import("./mn.3d5d1c3b.mjs"), "./locale/mr.js": () => import("./mr.ee8aa7ed.mjs"), "./locale/ms-my.js": () => import("./ms-my.620eda50.mjs"), "./locale/ms.js": () => import("./ms.818aff9b.mjs"), "./locale/mt.js": () => import("./mt.d87cebcd.mjs"), "./locale/my.js": () => import("./my.e64b60de.mjs"), "./locale/nb.js": () => import("./nb.36007e13.mjs"), "./locale/ne.js": () => import("./ne.b360711e.mjs"), "./locale/nl-be.js": () => import("./nl-be.09231ea4.mjs"), "./locale/nl.js": () => import("./nl.aa39d31c.mjs"), "./locale/nn.js": () => import("./nn.e7329ebf.mjs"), "./locale/oc-lnc.js": () => import("./oc-lnc.1fd3f8ad.mjs"), "./locale/pa-in.js": () => import("./pa-in.ca653b68.mjs"), "./locale/pl.js": () => import("./pl.31a8d98f.mjs"), "./locale/pt-br.js": () => import("./pt-br.be4382fe.mjs"), "./locale/pt.js": () => import("./pt.3e984855.mjs"), "./locale/rn.js": () => import("./rn.7e20dc01.mjs"), "./locale/ro.js": () => import("./ro.b5704f0f.mjs"), "./locale/ru.js": () => import("./ru.8ba4d876.mjs"), "./locale/rw.js": () => import("./rw.08ed5edb.mjs"), "./locale/sd.js": () => import("./sd.531016ab.mjs"), "./locale/se.js": () => import("./se.567090c3.mjs"), "./locale/si.js": () => import("./si.49090bd8.mjs"), "./locale/sk.js": () => import("./sk.5415cf1c.mjs"), "./locale/sl.js": () => import("./sl.ba536e05.mjs"), "./locale/sq.js": () => import("./sq.72d4d23e.mjs"), "./locale/sr-cyrl.js": () => import("./sr-cyrl.16f9bc68.mjs"), "./locale/sr.js": () => import("./sr.e8b2b0a2.mjs"), "./locale/ss.js": () => import("./ss.af1213b8.mjs"), "./locale/sv-fi.js": () => import("./sv-fi.85555e81.mjs"), "./locale/sv.js": () => import("./sv.497ec63e.mjs"), "./locale/sw.js": () => import("./sw.1e1cb2c1.mjs"), "./locale/ta.js": () => import("./ta.c0048b3e.mjs"), "./locale/te.js": () => import("./te.18d4c3b3.mjs"), "./locale/tet.js": () => import("./tet.20c06362.mjs"), "./locale/tg.js": () => import("./tg.040b7971.mjs"), "./locale/th.js": () => import("./th.dbbf0038.mjs"), "./locale/tk.js": () => import("./tk.aba5ca8d.mjs"), "./locale/tl-ph.js": () => import("./tl-ph.2831167b.mjs"), "./locale/tlh.js": () => import("./tlh.b54dab0c.mjs"), "./locale/tr.js": () => import("./tr.5d5dffaa.mjs"), "./locale/tzl.js": () => import("./tzl.c3ac2490.mjs"), "./locale/tzm-latn.js": () => import("./tzm-latn.7188d8d4.mjs"), "./locale/tzm.js": () => import("./tzm.2d707b38.mjs"), "./locale/ug-cn.js": () => import("./ug-cn.dcca5281.mjs"), "./locale/uk.js": () => import("./uk.69089d45.mjs"), "./locale/ur.js": () => import("./ur.e47fb659.mjs"), "./locale/uz-latn.js": () => import("./uz-latn.18cb977d.mjs"), "./locale/uz.js": () => import("./uz.d49342fb.mjs"), "./locale/vi.js": () => import("./vi.e541bbf3.mjs"), "./locale/x-pseudo.js": () => import("./x-pseudo.6ab6868d.mjs"), "./locale/yo.js": () => import("./yo.c21e5984.mjs"), "./locale/zh-cn.js": () => import("./zh-cn.bc2d9cc3.mjs"), "./locale/zh-hk.js": () => import("./zh-hk.2a870a67.mjs"), "./locale/zh-tw.js": () => import("./zh-tw.6d93d10f.mjs"), "./locale/zh.js": () => import("./zh.9889a73e.mjs") });
        for (const g in v)
          v[g]().then(() => {
            s.locale(d);
            let o, m;
            if (H()) {
              if (D()) {
                if (e.modelValue.length > 0) {
                  const [P, W] = e.modelValue;
                  o = s(P, e.formatter.date, !0), m = s(W, e.formatter.date, !0);
                }
              } else if (F()) {
                if (!Dt(e.modelValue))
                  try {
                    console.log(Object.keys(e.modelValue));
                  } catch {
                    console.warn(
                      "[Vue Tailwind Datepicker]: It looks like you want to use Object as the argument %cv-model",
                      "font-style: italic; color: #42b883;",
                      ", but you pass it undefined or null."
                    ), console.warn(
                      "[Vue Tailwind Datepicker]: We has replace with %c{ startDate: '', endDate: '' }",
                      "font-style: italic; color: #42b883;",
                      ", but you can replace manually."
                    ), a("update:modelValue", {
                      startDate: "",
                      endDate: ""
                    });
                  }
                if (e.modelValue) {
                  const [P, W] = Object.values(e.modelValue);
                  o = P && s(P, e.formatter.date, !0), m = W && s(W, e.formatter.date, !0);
                }
              } else if (e.modelValue) {
                const [P, W] = e.modelValue.split(e.separator);
                o = s(P, e.formatter.date, !0), m = s(W, e.formatter.date, !0);
              }
              o && m ? (y.value = f(
                {
                  previous: o,
                  next: m
                },
                e
              ), m.isBefore(o, "month") ? (r.value.previous = m, r.value.next = o, r.value.year.previous = m.year(), r.value.year.next = o.year()) : m.isSame(o, "month") ? (r.value.previous = o, r.value.next = m.add(1, "month"), r.value.year.previous = o.year(), r.value.year.next = o.add(1, "year").year()) : (r.value.previous = o, r.value.next = m, r.value.year.previous = o.year(), r.value.year.next = m.year()), e.autoApply || (x.value = [o, m])) : (r.value.previous = s(e.startFrom), r.value.next = s(e.startFrom).add(1, "month"), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year());
            } else {
              if (D()) {
                if (e.modelValue.length > 0) {
                  const [P] = e.modelValue;
                  o = s(P, e.formatter.date, !0);
                }
              } else if (F()) {
                if (e.modelValue) {
                  const [P] = Object.values(e.modelValue);
                  o = s(P, e.formatter.date, !0);
                }
              } else if (e.modelValue.length) {
                const [P] = e.modelValue.split(e.separator);
                o = s(P, e.formatter.date, !0);
              }
              o && o.isValid() ? (y.value = h(o, e), r.value.previous = o, r.value.next = o.add(1, "month"), r.value.year.previous = o.year(), r.value.year.next = o.add(1, "year").year(), e.autoApply || (x.value = [o])) : (r.value.previous = s(e.startFrom), r.value.next = s(e.startFrom).add(1, "month"), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year());
            }
            r.value.weeks = s.weekdaysShort(), r.value.months = e.formatter.month === "MMM" ? s.monthsShort() : s.months();
          }).catch((o) => {
            console.warn(
              o.message
            );
          });
      });
    });
    const pe = (d) => (d && A.value === null && (A.value = $(b.value)), d && A.value ? "place-right" : "place-left"), ve = (d) => (d && A.value === null && (A.value = $(b.value)), A.value ? "left-auto right-0" : "left-0 right-auto");
    return ie("isBetweenRange", z), ie("betweenRangeClasses", X), ie("datepickerClasses", q), ie("atMouseOver", T), ie("setToToday", ce), ie("setToYesterday", he), ie("setToLastDay", me), ie("setToThisMonth", fe), ie("setToLastMonth", re), ie("setToCustomShortcut", le), (d, v) => (K(), Ie(G(mt), {
      as: "div",
      id: "vtd",
      class: "relative w-full"
    }, {
      default: De(({ open: g }) => [
        e.overlay ? (K(), Ie(G(Mr), {
          key: 0,
          class: "fixed inset-0 bg-black opacity-30"
        })) : Oe("", !0),
        se(G(jr), {
          as: "label",
          class: "relative block"
        }, {
          default: De(() => [
            Ot(d.$slots, "default", {
              value: y.value,
              placeholder: Z.value,
              clear: M
            }, () => [
              ye(L("input", Vt({
                ref_key: "VtdInputRef",
                ref: w,
                type: "text",
                class: [
                  "relative block w-full",
                  t.inputClasses || "pl-3 pr-12 py-2.5 rounded-lg overflow-hidden border-solid text-sm text-vtd-secondary-700 placeholder-vtd-secondary-400 transition-colors bg-white border border-vtd-secondary-300 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700 dark:text-vtd-secondary-100 dark:placeholder-vtd-secondary-500 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-20"
                ],
                autocomplete: "off",
                "data-lpignore": "true",
                "data-form-type": "other"
              }, d.$attrs, {
                "onUpdate:modelValue": v[0] || (v[0] = (o) => y.value = o),
                placeholder: Z.value,
                onKeyup: E
              }), null, 16, Ar), [
                [Pt, y.value]
              ]),
              L("div", Lr, [
                L("button", {
                  type: "button",
                  class: "px-2 py-1 mr-1 focus:outline-none text-vtd-secondary-400 dark:text-opacity-70 rounded-md",
                  onClick: v[1] || (v[1] = (o) => y.value ? M() : d.$refs.VtdInputRef.focus())
                }, [
                  (K(), J("svg", Br, [
                    y.value ? (K(), J("path", Er)) : (K(), J("path", Fr))
                  ]))
                ])
              ])
            ])
          ]),
          _: 3
        }),
        se(tt, {
          "enter-from-class": "opacity-0 translate-y-3",
          "enter-to-class": "opacity-100 translate-y-0",
          "enter-active-class": "transform transition ease-out duration-200",
          "leave-active-class": "transform transition ease-in duration-150",
          "leave-from-class": "opacity-100 translate-y-0",
          "leave-to-class": "opacity-0 translate-y-3"
        }, {
          default: De(() => [
            se(G(Sr), {
              as: "div",
              class: "relative z-50"
            }, {
              default: De(({ close: o }) => [
                L("div", {
                  class: Pe(["absolute z-50 top-full sm:mt-2.5", ve(g)])
                }, [
                  L("div", {
                    ref_key: "VtdRef",
                    ref: b,
                    class: "fixed inset-0 z-50 overflow-y-auto sm:overflow-visible sm:static sm:z-auto bg-white dark:bg-vtd-secondary-800 sm:rounded-lg shadow-sm"
                  }, [
                    L("div", {
                      class: Pe(["vtd-datepicker static sm:relative w-full bg-white sm:rounded-lg sm:shadow-sm border-0 sm:border border-black/[.1] px-3 py-3 sm:px-4 sm:py-4 dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700/[1]", pe(g)])
                    }, [
                      L("div", Nr, [
                        e.shortcuts ? (K(), Ie(sr, {
                          key: 0,
                          shortcuts: e.shortcuts,
                          "as-range": H(),
                          "as-single": e.asSingle,
                          i18n: e.options.shortcuts,
                          close: o
                        }, null, 8, ["shortcuts", "as-range", "as-single", "i18n", "close"])) : Oe("", !0),
                        L("div", Hr, [
                          H() && !e.asSingle ? (K(), J("div", Rr, Ur)) : Oe("", !0),
                          L("div", {
                            class: Pe(["relative w-full sm:w-80", {
                              "mb-3 sm:mb-0 sm:mr-2": H() && !e.asSingle
                            }])
                          }, [
                            se(Ke, {
                              panel: u.previous,
                              calendar: G(_).previous
                            }, null, 8, ["panel", "calendar"]),
                            L("div", zr, [
                              ye(se(Je, {
                                months: G(R),
                                "onUpdate:month": G(_).previous.setMount
                              }, null, 8, ["months", "onUpdate:month"]), [
                                [xe, u.previous.month]
                              ]),
                              ye(se(Xe, {
                                years: G(_).previous.years(),
                                "onUpdate:year": G(_).previous.setYear
                              }, null, 8, ["years", "onUpdate:year"]), [
                                [xe, u.previous.year]
                              ]),
                              ye(L("div", null, [
                                se(qe, { weeks: G(S) }, null, 8, ["weeks"]),
                                se(Qe, {
                                  calendar: G(_).previous,
                                  weeks: G(S),
                                  "as-range": H(),
                                  "onUpdate:date": (m, P) => k(m, P, o)
                                }, null, 8, ["calendar", "weeks", "as-range", "onUpdate:date"])
                              ], 512), [
                                [xe, u.previous.calendar]
                              ])
                            ])
                          ], 2),
                          H() && !e.asSingle ? (K(), J("div", Wr, [
                            se(Ke, {
                              "as-prev-or-next": "",
                              panel: u.next,
                              calendar: G(_).next
                            }, null, 8, ["panel", "calendar"]),
                            L("div", Zr, [
                              ye(se(Je, {
                                months: G(R),
                                "onUpdate:month": G(_).next.setMount
                              }, null, 8, ["months", "onUpdate:month"]), [
                                [xe, u.next.month]
                              ]),
                              ye(se(Xe, {
                                "as-prev-or-next": "",
                                years: G(_).next.years(),
                                "onUpdate:year": G(_).next.setYear
                              }, null, 8, ["years", "onUpdate:year"]), [
                                [xe, u.next.year]
                              ]),
                              ye(L("div", null, [
                                se(qe, { weeks: G(S) }, null, 8, ["weeks"]),
                                se(Qe, {
                                  "as-prev-or-next": "",
                                  calendar: G(_).next,
                                  weeks: G(S),
                                  "as-range": H(),
                                  "onUpdate:date": (m, P) => k(m, P, o)
                                }, null, 8, ["calendar", "weeks", "as-range", "onUpdate:date"])
                              ], 512), [
                                [xe, u.next.calendar]
                              ])
                            ])
                          ])) : Oe("", !0)
                        ])
                      ]),
                      e.autoApply ? (K(), J("div", Qr, [
                        L("div", eo, [
                          L("div", to, [
                            L("button", {
                              type: "button",
                              onClick: (m) => o(),
                              class: "away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              textContent: ne(e.options.footer.cancel)
                            }, null, 8, ro)
                          ])
                        ])
                      ])) : (K(), J("div", Gr, [
                        L("div", Kr, [
                          L("div", Jr, [
                            L("button", {
                              type: "button",
                              class: "away-apply-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-vtd-primary-600 text-base font-medium text-white hover:bg-vtd-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800 disabled:cursor-not-allowed",
                              disabled: e.asSingle ? x.value.length < 1 : x.value.length < 2,
                              onClick: (m) => N(o),
                              textContent: ne(e.options.footer.apply)
                            }, null, 8, qr),
                            L("button", {
                              type: "button",
                              onClick: (m) => o(),
                              class: "mt-3 away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              textContent: ne(e.options.footer.cancel)
                            }, null, 8, Xr)
                          ])
                        ])
                      ]))
                    ], 2)
                  ], 512)
                ], 2)
              ]),
              _: 2
            }, 1024)
          ]),
          _: 2
        }, 1024)
      ]),
      _: 3
    }));
  }
}, wt = /* @__PURE__ */ (() => {
  const t = oo;
  return t.install = (a) => {
    a.component("VueTailwindDatepicker", t);
  }, t;
})(), ao = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: wt
}, Symbol.toStringTag, { value: "Module" }));
Object.entries(ao).forEach(([t, a]) => {
  t !== "default" && (wt[t] = a);
});
export {
  wt as c,
  s as d
};
